--- a/glibc.SlackBuild	2018-08-04 01:01:18.344067714 +0300
+++ b/glibc.SlackBuild	2018-09-23 13:15:17.050781031 +0300
@@ -95,6 +95,10 @@
     OPTIMIZ="-O3 -fPIC"
     LIBDIRSUFFIX="64"
     ;;
+  aarch64)
+    OPTIMIZ="-O3"
+    LIBDIRSUFFIX="64"
+    ;;
   *)
     OPTIMIZ="-O3"
     LIBDIRSUFFIX=""
@@ -109,6 +113,9 @@
     # This should be i586 for all 32-bit x86 arch:
     TARGET=${TARGET:-i586}
     ;;
+  aarch64)
+    TARGET=${TARGET:-aarch64}
+    ;;
 esac
 
 # Hand off the $ARCH variable to $SLACKWARE_ARCH to avoid confusing glibc:
@@ -130,6 +137,9 @@
 # This function fixes a doinst.sh file for x86_64.
 # With thanks to Fred Emmott.
 fix_doinst() {
+  # ARM uses .so.3:
+  sed -i 's?ld-linux.so.[0-9]?'"$ARMLDLINUXVER"'?g' install/doinst.sh
+
   if [ "x$LIBDIRSUFFIX" = "x" ]; then
     return;
   fi;
@@ -143,6 +153,12 @@
   if [ "$SLACKWARE_ARCH" = "x86_64" ]; then
     sed -i 's#ld-linux.so.2#ld-linux-x86-64.so.2#' install/doinst.sh
   fi
+
+  if [ "$SLACKWARE_ARCH" = "aarch64" ]; then
+    sed -i "/# Install NPTL glibc libraries:/i ( cd lib ; ln -sf ../lib${LIBDIRSUFFIX}/incoming/ld-${VERSION}.so $ARMLDLINUXVER )\n" install/doinst.sh
+    sed -i "/# Now, get rid of the temporary directory/i \\\n( cd lib ; ln -sf ../lib${LIBDIRSUFFIX}/ld-${VERSION}.so $ARMLDLINUXVER )\n" install/doinst.sh
+    sed -i "/# More links:/,${/cd lib ;.*$ARMLDLINUXVER/s/^/#/}" install/doinst.sh
+  fi
 }
 
 # This is a patch function to put all glibc patches in the build script
@@ -213,6 +229,12 @@
 # Clean up leftover CVS directories:
 find . -type d -name CVS -exec rm -r {} \; 2> /dev/null
 
+# Determine version of ld-linux.so
+case $SLACKWARE_ARCH in
+  arm)     export ARMLDLINUXVER="$( grep -A1 "hard-float ABI" sysdeps/unix/sysv/linux/arm/shlib-versions | egrep "^ld=" | awk -F= '{print $2}' )" ;;
+  aarch64) export ARMLDLINUXVER="$( egrep "^ld=" sysdeps/unix/sysv/linux/aarch64/shlib-versions | grep -v "be" | awk -F= '{print $2}' )" ;;
+esac
+
 # Apply patches; exit if any fail.
 apply_patches
 if [ ! $? = 0 ]; then
