--- a/glibc.SlackBuild	2019-05-31 23:20:56.663266631 +0300
+++ b/glibc.SlackBuild	2019-05-31 23:21:32.591818492 +0300
@@ -95,6 +95,10 @@
     OPTIMIZ="-O3 -fPIC"
     LIBDIRSUFFIX="64"
     ;;
+  aarch64)
+    OPTIMIZ="-O3"
+    LIBDIRSUFFIX="64"
+    ;;
   *)
     OPTIMIZ="-O3"
     LIBDIRSUFFIX=""
@@ -109,6 +113,9 @@
     # This should be i586 for all 32-bit x86 arch:
     TARGET=${TARGET:-i586}
     ;;
+  aarch64)
+    TARGET=${TARGET:-aarch64}
+    ;;
 esac
 
 # Hand off the $ARCH variable to $SLACKWARE_ARCH to avoid confusing glibc:
@@ -130,6 +137,9 @@
 # This function fixes a doinst.sh file for x86_64.
 # With thanks to Fred Emmott.
 fix_doinst() {
+  # ARM uses .so.3:
+  sed -i 's?ld-linux.so.[0-9]?'"$ARMLDLINUXVER"'?g' install/doinst.sh
+
   if [ "x$LIBDIRSUFFIX" = "x" ]; then
     return;
   fi;
@@ -143,6 +153,11 @@
   if [ "$SLACKWARE_ARCH" = "x86_64" ]; then
     sed -i 's#ld-linux.so.2#ld-linux-x86-64.so.2#' install/doinst.sh
   fi
+
+  if [ "$SLACKWARE_ARCH" = "aarch64" ]; then
+    sed -i "/# Install NPTL glibc libraries:/i ( cd lib ; ln -sf ../lib${LIBDIRSUFFIX}/incoming/ld-${VERSION}.so $ARMLDLINUXVER )\n" install/doinst.sh
+    sed -i "/# Now, get rid of the temporary directory/i \\\n( cd lib ; ln -sf ../lib${LIBDIRSUFFIX}/ld-${VERSION}.so $ARMLDLINUXVER )\n" install/doinst.sh
+  fi
 }
 
 # This is a patch function to put all glibc patches in the build script
@@ -215,6 +230,12 @@
 # Clean up leftover CVS directories:
 find . -type d -name CVS -exec rm -r {} \; 2> /dev/null
 
+# Determine version of ld-linux.so
+case $SLACKWARE_ARCH in
+  arm)     export ARMLDLINUXVER="$( grep -A1 "hard-float ABI" sysdeps/unix/sysv/linux/arm/shlib-versions | egrep "^ld=" | awk -F= '{print $2}' )" ;;
+  aarch64) export ARMLDLINUXVER="$( egrep "^ld=" sysdeps/unix/sysv/linux/aarch64/shlib-versions | grep -v "be" | awk -F= '{print $2}' )" ;;
+esac
+
 # Apply patches; exit if any fail.
 apply_patches
 if [ ! $? = 0 ]; then
@@ -243,6 +264,8 @@
   --with-tls \
   --with-__thread \
   --without-cvs \
+  libc_cv_rtlddir=/lib${LIBDIRSUFFIX} \
+  libc_cv_slibdir=/lib${LIBDIRSUFFIX} \
   $TARGET-slackware-linux
 
 make $NUMJOBS || make || exit 1
@@ -431,7 +454,7 @@
 # Call the function to fix doinst.sh where $LIBDIRSUFFIX is needed:
 fix_doinst
 # Only scrub the links in /lib{,64} that will be created by ldconfig:
-find lib${LIBDIRSUFFIX} -type l -exec rm {} \;
+find lib{,${LIBDIRSUFFIX}} -type l -exec rm {} \;
 # Build the package:
 makepkg -l y -c n $TMP/glibc-solibs-$VERSION-$SLACKWARE_ARCH-$BUILD.txz
 
@@ -444,7 +467,7 @@
 chown -R root:root etc
 chmod 755 etc/profile.d/*
 # Only scrub the links in /lib{,64} that will be created by ldconfig:
-find lib${LIBDIRSUFFIX} -type l -exec rm {} \;
+find lib{,${LIBDIRSUFFIX}} -type l -exec rm {} \;
 mkdir install
 cp -a $CWD/slack-desc.glibc install/slack-desc
 cp -a $CWD/doinst.sh-glibc install/doinst.sh
@@ -466,3 +489,4 @@
 # Done!
 echo
 echo "glibc packages built in $TMP!"
+
