--- a/linuxdoc-tools.build	2019-02-05 16:55:28.000000000 +0000
+++ b/linuxdoc-tools.build	2019-02-06 18:38:50.767601088 +0000
@@ -108,7 +108,8 @@
   --prefix=/usr \
   --sysconfdir=/etc \
   --docdir=/usr/doc/asciidoc-$ASCIIDOCVER \
-  --mandir=/usr/man || exit 1
+  --mandir=/usr/man \
+  --build=$ARCH-unknown-linux$HOSTTARGET || exit 1
 
 # Install:
 if [ "$1" = "nodocs" ]; then
@@ -346,6 +347,7 @@
   --enable-http \
   --enable-default-catalog=/etc/sgml/catalog \
   --enable-default-search-path=/usr/share/sgml \
+  --build=$ARCH-unknown-linux$HOSTTARGET \
   || exit 1
 
 # Build:
@@ -426,6 +428,7 @@
   --enable-default-catalog=/etc/sgml/catalog \
   --enable-default-search-path=/usr/share/sgml \
   --datadir=/usr/share/sgml/openjade-$OPENJADEVER \
+  --build=$ARCH-unknown-linux$HOSTTARGET \
   || exit 1
 
 # Build:
@@ -1040,7 +1043,7 @@
   --mandir=/usr/man \
   --disable-scrollkeeper \
   --enable-build-utils \
-  --build=$ARCH-slackware-linux$HOSTTARGET || exit 1
+  --build=$ARCH-unknown-linux$HOSTTARGET || exit 1
 
 # No parallel build here because it doesn't work:
 make || exit 1
@@ -1168,7 +1171,7 @@
    --disable-dependency-tracking \
    --program-transform-name="s/^docbook2/docbook2x-/" \
    --with-html-xsl \
-   --build=$ARCH-slackware-linux$HOSTTARGET || exit 1
+   --build=$ARCH-unknown-linux$HOSTTARGET || exit 1
 
 # Build:
 make || exit 1
