--- a/scim-anthy.SlackBuild	2017-10-01 20:47:36.393144397 +0100
+++ b/scim-anthy.SlackBuild	2017-10-01 20:57:24.770747191 +0100
@@ -58,6 +58,9 @@
 elif [ "$ARCH" = "x86_64" ]; then
   SLKCFLAGS="-O2 -fPIC"
   LIBDIRSUFFIX="64"
+elif [ "$ARCH" = "aarch64" ]; then
+  SLKCFLAGS="-O2"
+  LIBDIRSUFFIX="64"
 fi
 
 CWD=$(pwd)
@@ -72,6 +75,7 @@
 cd ${PKGNAM}-${VERSION} || exit 1
 
 zcat $CWD/scim-anthy.gcc43.diff.gz | patch -p1 --verbose || exit 1
+xzcat $CWD/scim-anthy-fix-typo.patch.xz | patch -p1 --verbose || exit 1
 
 chown -R root:root .
 find . \
