--- a/cups.SlackBuild	2018-03-31 01:27:42.680440201 +0300
+++ b/cups.SlackBuild	2018-03-31 01:30:43.718440201 +0300
@@ -61,6 +61,9 @@
 elif [ "$ARCH" = "armv7hl" ]; then
   SLKCFLAGS="-O3 -march=armv7-a -mfpu=vfpv3-d16"
   LIBDIRSUFFIX=""
+elif [ "$ARCH" = "aarch64" ]; then
+  SLKCFLAGS="-O2"
+  LIBDIRSUFFIX="64"
 else
   SLKCFLAGS="-O2"
   LIBDIRSUFFIX=""
@@ -84,6 +87,8 @@
 
 sed -i.orig -e 's#$exec_prefix/lib/cups#$libdir/cups#g' configure
 
+CC=gcc \
+CXX=g++ \
 CFLAGS="$SLKCFLAGS" \
 CXXFLAGS="$SLKCFLAGS" \
 ./configure \
