--- a/gdb.SlackBuild	2017-09-21 21:03:35.256660562 +0300
+++ b/gdb.SlackBuild	2017-09-21 21:04:16.720660562 +0300
@@ -25,7 +25,7 @@
 VERSION=${VERSION:-$(echo $PKGNAM-*.tar.xz | rev | cut -f 3- -d . | cut -f 1 -d - | rev)}
 BUILD=${BUILD:-1}
 
-NUMJOBS=${NUMJOBS:-" -j7 "}
+NUMJOBS=${NUMJOBS:-" -j5 "}
 
 # Automatically determine the architecture we're building on:
 if [ -z "$ARCH" ]; then
@@ -37,11 +37,10 @@
   esac
 fi
 
-if [ "$ARCH" = "x86_64" ]; then
-  LIBDIRSUFFIX="64"
-else
-  LIBDIRSUFFIX=""
-fi
+case "$ARCH" in
+  x86_64|aarch64) LIBDIRSUFFIX="64" ;;
+               *) LIBDIRSUFFIX="" ;;
+esac
 
 CWD=$(pwd)
 TMP=${TMP:-/tmp}
