--- a/links.SlackBuild	1970-01-14 22:10:30.395676712 +0100
+++ b/links.SlackBuild	1970-01-14 22:10:53.220970207 +0100
@@ -72,6 +72,8 @@
  \( -perm 666 -o -perm 664 -o -perm 600 -o -perm 444 -o -perm 440 -o -perm 400 \) \
  -exec chmod 644 {} \;
 
+sed -i "/ac_cpp=/s/\$CPPFLAGS/\$CPPFLAGS -O2/" configure
+
 # We are configuring this without X as there are MANY browser choices for
 # X already.  If you need a browser for X, either use something else or
 # recompile.  :-)
