--- a/qt5-webkit.SlackBuild	2023-05-04 06:21:34.811633245 +0300
+++ b/qt5-webkit.SlackBuild	2023-07-02 01:15:14.776995464 +0300
@@ -63,6 +63,13 @@ elif [ "$ARCH" = "s390" ]; then
 elif [ "$ARCH" = "x86_64" ]; then
   SLKCFLAGS="-O2 -fPIC"
   LIBDIRSUFFIX="64"
+elif [ "$ARCH" = "aarch64" ]; then
+  SLKCFLAGS="-O2 -fPIC"
+  LIBDIRSUFFIX="64"
+elif [ "$ARCH" = "riscv64" ]; then
+  SLKCFLAGS="-O2 -fPIC"
+  LIBDIRSUFFIX="64"
+  CMAKE_EXTRA="-DCMAKE_EXE_LINKER_FLAGS=-latomic -DCMAKE_SHARED_LINKER_FLAGS=-latomic -DENABLE_JIT=OFF"
 elif [ "$ARCH" = "armv7hl" ]; then
   SLKCFLAGS="-O3 -march=armv7-a -mfpu=vfpv3-d16"
   LIBDIRSUFFIX=""
@@ -109,6 +116,13 @@ zcat $CWD/webkit-offlineasm-warnings-rub
 # Fix for gcc13:
 zcat $CWD/qt5-webkit.gcc13.diff.gz | patch -p1 --verbose || exit 1
 
+# Add new architectures (riscv64, or1k, sparc64, loongarch64):
+zcat $CWD/new_architectures.diff.gz | patch -p1 --verbose || exit 1
+
+if [ ! -z "$SLKLDFLAGS" ]; then
+  export LDFLAGS="$SLKLDFLAGS"
+fi
+
 # -DU_DEFINE_FALSE_AND_TRUE=1 since recent icu4c no longer defines these otherwise.
 
 export CFLAGS="$SLKCFLAGS -DU_DEFINE_FALSE_AND_TRUE=1"
@@ -121,6 +135,7 @@ export LD_LIBRARY_PATH="${QTDIR}/qtbase/
 mkdir qtwebkit-build
 cd qtwebkit-build
   cmake \
+    $CMAKE_EXTRA \
     -DPORT=Qt \
     -DQt5_DIR=$QTDIR \
     ..  || exit 1
