--- a/llvm.SlackBuild	2019-06-09 12:40:01.492506323 +0100
+++ b/llvm.SlackBuild	2019-06-09 12:44:38.040650311 +0100
@@ -59,6 +59,9 @@
 elif [ "$ARCH" = "x86_64" ]; then
   SLKCFLAGS="-O2 -fPIC"
   LIBDIRSUFFIX="64"
+elif [ "$ARCH" = "aarch64" ]; then
+  SLKCFLAGS="-O2"
+  LIBDIRSUFFIX="64"
 else
   SLKCFLAGS="-O2"
   LIBDIRSUFFIX=""
@@ -112,7 +115,7 @@
 # clangd to avoid the failure. Any help with a proper fix is appreciated.
 # I have not reported this upstream since I suspect that it's an artifact
 # of our convoluted build method.
-if [ $ARCH = i586 ] || [ $ARCH = i686 ]; then
+if [ $ARCH = i?86 ] || [ $ARCH = aarch64 ]; then
   zcat $CWD/clang-tools-extra.no.clangd.diff.gz | patch -p1 --verbose --backup --suffix=.orig || exit 1
 fi
 
@@ -127,13 +130,29 @@
 # by building clang with shared libs
 zcat $CWD/n_clang_allow_BUILD_SHARED_LIBRARY.patch.gz | patch -p0 --verbose || exit 1
 
+xzcat $CWD/clang.dynamic.linker.aarch64.patch.xz | patch -p1 --verbose || exit 1
+xzcat $CWD/clang.toolchains.aarch64.triple.patch.xz | patch -p1 --verbose || exit 1
+
+export SLKCFLAGS="$SLKCFLAGS -fPIC -Wall -Wp,-D_FORTIFY_SOURCE=2 -fpermissive -fexceptions -fstack-protector-strong --param=ssp-buffer-size=4 -fuse-ld=gold"
+export SLKCONFARGS="-DLLVM_HOST_TRIPLE=$ARCH-slackware-linux \
+                    -DCMAKE_VERBOSE_MAKEFILE:BOOL=ON"
+export LDFLAGS="-Wl,-z,relro -Wl,--as-needed -Wl,-z,now"
+
+CCOMPILER=gcc
+CPPCOMPILER=g++
+which clang > /dev/null 2>&1 && { CCOMPILER=clang ; CPPCOMPILER=clang++ ; SLKCFLAGS="${SLKCFLAGS%-fuse-ld*}" ; }
+echo "Using compiler: $CCOMPILER / $CPPCOMPILER"
+
 # need to disable assertions to make llvm thread-safe
 # clang resource dir is a relative path based on the location of the clang binary
 mkdir build
 cd build
   cmake \
-    -DCMAKE_C_COMPILER="clang" \
-    -DCMAKE_CXX_COMPILER="clang++" \
+    \
+    $SLKCONFARGS \
+    \
+    -DCMAKE_C_COMPILER="$CCOMPILER" \
+    -DCMAKE_CXX_COMPILER="$CPPCOMPILER" \
     -DCMAKE_C_FLAGS:STRING="$SLKCFLAGS" \
     -DCMAKE_CXX_FLAGS:STRING="$SLKCFLAGS" \
     -DCMAKE_INSTALL_PREFIX=/usr \
