--- a/qt5.SlackBuild	2020-02-15 08:05:12.178060484 +0000
+++ b/qt5.SlackBuild	2020-02-15 08:18:19.955536251 +0000
@@ -78,6 +78,11 @@
   SLKCFLAGS="-O2 -march=armv7-a -mfpu=vfpv3-d16 -fno-strict-volatile-bitfields"
   SLKLDFLAGS=""
   LIBDIRSUFFIX=""
+elif [ "$ARCH" = "aarch64" ]; then
+  SLKCFLAGS="-O2"
+  SLKLDFLAGS=""
+  SLKLDFLAGS="-L/usr/lib64"
+  LIBDIRSUFFIX="64"
 else
   SLKCFLAGS="-O2"
   SLKLDFLAGS=""
@@ -150,6 +155,19 @@
   PACONF=" "
 fi
 
+if [ "$ARCH" == "aarch64" ]; then
+  EXTRACONF="-no-sse2 \
+             -no-sse3 \
+             -no-ssse3 \
+             -no-sse4.1 \
+             -no-sse4.2 \
+             -no-avx \
+             -no-avx2 \
+             -no-feature-getentropy"
+else
+  EXTRACONF=" "
+fi
+
 # Use our custom compiler and linker flags:
 sed -e "s|^\(QMAKE_CFLAGS_RELEASE.*\)|\1 ${SLKCFLAGS}|" \
   -i qtbase/mkspecs/common/gcc-base.conf || exit 1
@@ -202,6 +220,7 @@
   -webengine-proprietary-codecs \
   ${PACONF} \
   ${RELOCATIONS} \
+  ${EXTRACONF} \
   -no-pch \
   # No-precompiled-headers is ccache-friendly.
   
