--- a/qt5.SlackBuild	2021-02-13 03:46:31.830669992 +0200
+++ b/qt5.SlackBuild	2021-02-13 03:47:40.154632038 +0200
@@ -81,6 +81,16 @@
   SLKCFLAGS="-O2 -march=armv7-a -mfpu=vfpv3-d16 -fno-strict-volatile-bitfields"
   SLKLDFLAGS=""
   LIBDIRSUFFIX=""
+elif [ "$ARCH" = "aarch64" ]; then
+  SLKCFLAGS="-O2 -fPIC"
+  SLKLDFLAGS=""
+  SLKLDFLAGS="-L/usr/lib64"
+  LIBDIRSUFFIX="64"
+elif [ "$ARCH" = "riscv64" ]; then
+  SLKCFLAGS="-O2 -fPIC"
+  SLKLDFLAGS=""
+  SLKLDFLAGS="-L/usr/lib64"
+  LIBDIRSUFFIX="64"
 else
   SLKCFLAGS="-O2"
   SLKLDFLAGS=""
@@ -146,6 +156,19 @@
   unset TAG
 fi
 
+if [ "$ARCH" == "aarch64" || "$ARCH" == "riscv64" ]; then
+  EXTRACONF="-no-sse2 \
+             -no-sse3 \
+             -no-ssse3 \
+             -no-sse4.1 \
+             -no-sse4.2 \
+             -no-avx \
+             -no-avx2 \
+             -no-feature-getentropy"
+else
+  EXTRACONF=" "
+fi
+
 # Use our custom compiler and linker flags:
 sed -e "s|^\(QMAKE_CFLAGS_RELEASE.*\)|\1 ${SLKCFLAGS}|" \
   -i qtbase/mkspecs/common/gcc-base.conf || exit 1
@@ -210,6 +233,7 @@
   ${FFMPEG_OPTION} \
   ${PULSEAUDIO_OPTION} \
   ${RELOCATIONS} \
+  ${EXTRACONF} \
   -no-pch
   # No-precompiled-headers is ccache-friendly.
   
