--- a/qt5.SlackBuild	2023-07-01 13:38:40.212964533 +0300
+++ b/qt5.SlackBuild	2023-07-01 19:52:37.795974868 +0300
@@ -81,6 +81,14 @@ elif [ "$ARCH" = "armv7hl" ]; then
   SLKCFLAGS="-O2 -march=armv7-a -mfpu=vfpv3-d16 -fno-strict-volatile-bitfields"
   SLKLDFLAGS=""
   LIBDIRSUFFIX=""
+elif [ "$ARCH" = "aarch64" ]; then
+  SLKCFLAGS="-O2 -fPIC"
+  SLKLDFLAGS="-L/usr/lib64"
+  LIBDIRSUFFIX="64"
+elif [ "$ARCH" = "riscv64" ]; then
+  SLKCFLAGS="-O2 -fPIC"
+  SLKLDFLAGS="-L/usr/lib64 -latomic"
+  LIBDIRSUFFIX="64"
 else
   SLKCFLAGS="-O2"
   SLKLDFLAGS=""
@@ -133,6 +141,16 @@ cd qtwebsockets
 cd - 1>/dev/null
 cd qtwebengine
   zcat $CWD/patches/qt5-qtwebengine-gcc11.patch.gz | patch -p1 --verbose || exit 1
+  zcat $CWD/patches/qtwebengine-mips-ppc-riscv-s390-support.diff.gz | patch -p1 --verbose || exit 1
+  zcat $CWD/patches/qtwebengine-mips32-limit-mksnapshot-memory.diff.gz | patch -p1 --verbose || exit 1
+  zcat $CWD/patches/qtwebengine-bigendian.diff.gz | patch -p1 --verbose || exit 1
+  zcat $CWD/patches/qtwebengine-time64_disable_zlib.diff.gz | patch -p1 --verbose || exit 1
+  tar xvf $CWD/patches/qtwebengine-mips-ppc-riscv-s390-newfiles.tar.gz
+  if [ "${ARCH::3}" = "ppc" -o "$ARCH" = "mips64el" -o "$ARCH" = "riscv64" -o "$ARCH" = "s390x" ]
+  then
+    # Disable getcontext on breakpad (missing assembler):
+    zcat $CWD/patches/qtwebengine-nogetcontext.diff.gz | patch -p1 --verbose || exit 1
+  fi
 cd - 1>/dev/null
 
 # Fix build with gcc12:
