--- a/build-deps.sh	2022-12-17 23:42:56.562587644 +0000
+++ b/build-deps.sh	2022-12-17 23:46:37.664198318 +0000
@@ -27,8 +27,8 @@ mkdir -p $TMP/mozilla-thunderbird-build-
 # happening in /tmp:
 chmod 700 $TMP/mozilla-thunderbird-build-deps
 
-if /bin/ls build-deps*.txz 1> /dev/null 2> /dev/null ; then # use prebuilt
-  ( cd $TMP/mozilla-thunderbird-build-deps ; tar xf $CWD/build-deps*.txz )
+if /bin/ls build-deps*${ARCH}*.txz 1> /dev/null 2> /dev/null ; then # use prebuilt
+  ( cd $TMP/mozilla-thunderbird-build-deps ; tar xf $CWD/build-deps*${ARCH}*.txz )
 else
   # We need to use the incredibly ancient autoconf-2.13 for this  :/
   ( cd $CWD/build-deps/autoconf ; ./autoconf.build ) || exit 1
