--- a/mozilla-thunderbird.SlackBuild	2022-12-15 22:32:01.461875851 +0200
+++ b/mozilla-thunderbird.SlackBuild	2023-04-14 23:22:07.863127275 +0300
@@ -44,6 +44,9 @@ MOZLOCALIZE=${MOZLOCALIZE:-}
 # "UnicodeDecodeError: 'ascii' codec can't decode byte 0xe2 in position 36: ordinal not in range(128)"
 LANG=C
 
+# LTO needs more open files
+ulimit -n 4096
+
 # Add a shell script to start the thunderbird binary with MOZ_ALLOW_DOWNGRADE=1
 # to avoid backing up (and disabling) the user profile if a browser downgrade
 # is detected. If you want to build with the stock default behavior, set
@@ -101,6 +104,12 @@ elif [ "$ARCH" = "arm" ]; then
 elif [ "$ARCH" = "armel" ]; then
   SLKCFLAGS="-g0 -march=armv4t"
   LIBDIRSUFFIX=""
+elif [ "$ARCH" = "aarch64" ]; then
+  SLKCFLAGS="-g0 -fPIC"
+  LIBDIRSUFFIX="64"
+elif [ "$ARCH" = "riscv64" ]; then
+  SLKCFLAGS="-g0 -fPIC"
+  LIBDIRSUFFIX="64"
 else
   SLKCFLAGS="-g0"
   LIBDIRSUFFIX=""
@@ -194,6 +203,10 @@ fi
 case "$ARCH" in
     armv7hl) ARCH_CONFIG="--with-arch=armv7-a --with-float-abi=hard --with-fpu=vfpv3-d16 --disable-elf-hack"
              ;;
+    aarch64) ARCH_CONFIG="--disable-webrtc"
+             ;;
+    riscv64) ARCH_CONFIG="--disable-webrtc"
+             ;;
     *)       ARCH_CONFIG=" "
              ;;
 esac
@@ -207,6 +220,7 @@ find . \
 
 # Our building options, in a configure-like display ;)
 OPTIONS="\
+  $ARCH_CONFIG \
   --enable-official-branding \
   --prefix=/usr \
   --libdir=/usr/lib${LIBDIRSUFFIX} \
@@ -229,7 +243,6 @@ OPTIONS="\
   --disable-crashreporter \
   --disable-debug-symbols \
   --disable-debug \
-  --disable-elf-hack \
   --disable-tests \
   --disable-updater \
   --host=$ARCH-slackware-linux \
@@ -253,8 +266,9 @@ export MACH_USE_SYSTEM_PYTHON="1"
 export BUILD_OFFICIAL=1
 export MOZILLA_OFFICIAL=1
 export MOZ_PHOENIX=1
+export MOZ_PACKAGE_JSSHELL=1
 export CFLAGS="$SLKCFLAGS"
-export CXXFLAGS="$SLKCFLAGS"
+export CXXFLAGS="$SLKCFLAGS -fno-delete-null-pointer-checks"
 export MOZ_MAKE_FLAGS="$NUMJOBS"
 # Set the following variable to empty. =0 does not work.
 export MOZ_REQUIRE_SIGNING=
