--- a/mozilla-thunderbird.SlackBuild	2021-02-13 00:50:39.655701504 +0200
+++ b/mozilla-thunderbird.SlackBuild	2021-02-13 00:52:10.771602399 +0200
@@ -107,6 +107,14 @@
   SLKCFLAGS="-march=armv4 -mtune=xscale"
   LIBDIRSUFFIX=""
   OPTIMIZE=${OPTIMIZE:-"-O2 -g0"}
+elif [ "$ARCH" = "aarch64" ]; then
+  SLKCFLAGS="-fPIC"
+  LIBDIRSUFFIX="64"
+  OPTIMIZE=${OPTIMIZE:-"-O2 -g0"}
+elif [ "$ARCH" = "riscv64" ]; then
+  SLKCFLAGS="-fPIC"
+  LIBDIRSUFFIX="64"
+  OPTIMIZE=${OPTIMIZE:-"-O2 -g0"}
 else
   SLKCFLAGS=""
   LIBDIRSUFFIX=""
@@ -190,6 +198,14 @@
              # Make Thunderbird compile on ARM platforms lacking neon support:
              zcat $CWD/mozilla-firefox.xpcom_arm.patch.gz | patch -p1 --verbose || exit 1
              ;;
+    aarch64) ARCH_CONFIG="--disable-elf-hack"
+             xzcat $CWD/arm.patch.xz | patch -p1 --verbose || exit 1
+             xzcat $CWD/Bug-1238661---fix-mozillaSignalTrampoline-to-work-.patch.xz | patch -p1 --verbose || exit 1
+             ;;
+    riscv64) ARCH_CONFIG="--disable-elf-hack"
+             xzcat $CWD/arm.patch.xz | patch -p1 --verbose || exit 1
+             xzcat $CWD/Bug-1238661---fix-mozillaSignalTrampoline-to-work-.patch.xz | patch -p1 --verbose || exit 1
+             ;;
     *)       ARCH_CONFIG=" "
              ;;
 esac
@@ -275,6 +291,12 @@
 # Add the $OPTIONS above to .mozconfig:
 for option in $OPTIONS; do echo "ac_add_options $option" >> .mozconfig; done
 
+# ARM: see https://bugzilla.redhat.com/show_bug.cgi?id=1426850
+# This isn't added using $SLKCONFARGS because it doesn't cater for things with spaces in:
+# unless I edit the for loop above to use IFS with some other variable.
+# this is easier right now ;-)
+echo 'ac_add_options --enable-optimize="${OPTIMIZE} -fno-schedule-insns"' >> .mozconfig
+
 # Disable tests on standard build:
 echo "ac_add_options --disable-tests" >> .mozconfig
 ./mach build || exit 1
