--- a/mozilla-thunderbird.SlackBuild	2021-03-04 19:52:24.132870010 +0200
+++ b/mozilla-thunderbird.SlackBuild	2021-03-04 20:12:23.134624607 +0200
@@ -44,6 +44,9 @@
 # "UnicodeDecodeError: 'ascii' codec can't decode byte 0xe2 in position 36: ordinal not in range(128)"
 LANG=C
 
+# LTO needs more open files
+ulimit -n 4096
+
 # Add a shell script to start the thunderbird binary with MOZ_ALLOW_DOWNGRADE=1
 # to avoid backing up (and disabling) the user profile if a browser downgrade
 # is detected. If you want to build with the stock default behavior, set
@@ -107,6 +110,14 @@
   SLKCFLAGS="-march=armv4 -mtune=xscale"
   LIBDIRSUFFIX=""
   OPTIMIZE=${OPTIMIZE:-"-O2 -g0"}
+elif [ "$ARCH" = "aarch64" ]; then
+  SLKCFLAGS="-O2 -g0 -fPIC"
+  LIBDIRSUFFIX="64"
+  OPTIMIZE=${OPTIMIZE:-"-O2"}
+elif [ "$ARCH" = "riscv64" ]; then
+  SLKCFLAGS="-O2 -g0 -fPIC"
+  LIBDIRSUFFIX="64"
+  OPTIMIZE=${OPTIMIZE:-"-O2"}
 else
   SLKCFLAGS=""
   LIBDIRSUFFIX=""
@@ -187,6 +198,14 @@
              # Make Thunderbird compile on ARM platforms lacking neon support:
              zcat $CWD/mozilla-firefox.xpcom_arm.patch.gz | patch -p1 --verbose || exit 1
              ;;
+    aarch64) ARCH_CONFIG="--disable-webrtc"
+             xzcat $CWD/arm.patch.xz | patch -p1 --verbose || exit 1
+             xzcat $CWD/Bug-1238661---fix-mozillaSignalTrampoline-to-work-.patch.xz | patch -p1 --verbose || exit 1
+             ;;
+    riscv64) ARCH_CONFIG="--disable-webrtc"
+             xzcat $CWD/arm.patch.xz | patch -p1 --verbose || exit 1
+             xzcat $CWD/Bug-1238661---fix-mozillaSignalTrampoline-to-work-.patch.xz | patch -p1 --verbose || exit 1
+             ;;
     *)       ARCH_CONFIG=" "
              ;;
 esac
@@ -200,6 +219,7 @@
 
 # Our building options, in a configure-like display ;)
 OPTIONS="\
+  $ARCH_CONFIG \
   --enable-official-branding \
   --prefix=/usr \
   --libdir=/usr/lib${LIBDIRSUFFIX} \
@@ -216,7 +236,7 @@
   $GOOGLE_API_KEY \
   --disable-crashreporter \
   --disable-debug \
-  --disable-tests \
+  --disable-debug-symbols \
   --disable-updater \
   --host=$ARCH-slackware-linux \
   --target=$ARCH-slackware-linux"
@@ -238,8 +258,9 @@
 export BUILD_OFFICIAL=1
 export MOZILLA_OFFICIAL=1
 export MOZ_PHOENIX=1
+export MOZ_PACKAGE_JSSHELL=1
 export CFLAGS="$SLKCFLAGS"
-export CXXFLAGS="$SLKCFLAGS"
+export CXXFLAGS="$SLKCFLAGS -fno-delete-null-pointer-checks"
 export MOZ_MAKE_FLAGS="$NUMJOBS"
 
 # Clear some variables that could break the build
@@ -261,7 +282,7 @@
 mkdir -p mozilla/obj
 
 # Set options for $OPTIMIZE:
-echo "ac_add_options --enable-optimize=\"${OPTIMIZE}\"" >> .mozconfig
+echo 'ac_add_options --enable-optimize="${OPTIMIZE} -fno-schedule-insns"' >> .mozconfig
 
 if [ "$COMPILE_X86_UNDER_X86_64" = "true" ]; then
   # Compile for i686 under an x86_64 kernel:
