--- a/mozilla-thunderbird.SlackBuild	2021-03-25 18:42:51.255245522 +0200
+++ b/mozilla-thunderbird.SlackBuild	2021-03-25 18:58:13.257778314 +0200
@@ -44,6 +44,9 @@
 # "UnicodeDecodeError: 'ascii' codec can't decode byte 0xe2 in position 36: ordinal not in range(128)"
 LANG=C
 
+# LTO needs more open files
+ulimit -n 4096
+
 # Add a shell script to start the thunderbird binary with MOZ_ALLOW_DOWNGRADE=1
 # to avoid backing up (and disabling) the user profile if a browser downgrade
 # is detected. If you want to build with the stock default behavior, set
@@ -109,6 +112,12 @@
 elif [ "$ARCH" = "armel" ]; then
   SLKCFLAGS="-g0 -march=armv4t"
   LIBDIRSUFFIX=""
+elif [ "$ARCH" = "aarch64" ]; then
+  SLKCFLAGS="-g0 -fPIC"
+  LIBDIRSUFFIX="64"
+elif [ "$ARCH" = "riscv64" ]; then
+  SLKCFLAGS="-g0 -fPIC"
+  LIBDIRSUFFIX="64"
 else
   SLKCFLAGS="-g0"
   LIBDIRSUFFIX=""
@@ -195,6 +204,14 @@
              # Make Thunderbird compile on ARM platforms lacking neon support:
              zcat $CWD/mozilla-firefox.xpcom_arm.patch.gz | patch -p1 --verbose || exit 1
              ;;
+    aarch64) ARCH_CONFIG="--disable-webrtc"
+             xzcat $CWD/arm.patch.xz | patch -p1 --verbose || exit 1
+             xzcat $CWD/Bug-1238661---fix-mozillaSignalTrampoline-to-work-.patch.xz | patch -p1 --verbose || exit 1
+             ;;
+    riscv64) ARCH_CONFIG="--disable-webrtc"
+             xzcat $CWD/arm.patch.xz | patch -p1 --verbose || exit 1
+             xzcat $CWD/Bug-1238661---fix-mozillaSignalTrampoline-to-work-.patch.xz | patch -p1 --verbose || exit 1
+             ;;
     *)       ARCH_CONFIG=" "
              ;;
 esac
@@ -208,6 +225,7 @@
 
 # Our building options, in a configure-like display ;)
 OPTIONS="\
+  $ARCH_CONFIG \
   --enable-official-branding \
   --prefix=/usr \
   --libdir=/usr/lib${LIBDIRSUFFIX} \
@@ -223,12 +241,10 @@
   --disable-install-strip \
   --enable-cpp-rtti \
   --enable-accessibility \
-  --enable-optimize \
   $GOOGLE_API_KEY \
   --disable-crashreporter \
   --disable-debug-symbols \
   --disable-debug \
-  --disable-elf-hack \
   --disable-tests \
   --disable-updater \
   --host=$ARCH-slackware-linux \
@@ -252,8 +268,9 @@
 export BUILD_OFFICIAL=1
 export MOZILLA_OFFICIAL=1
 export MOZ_PHOENIX=1
+export MOZ_PACKAGE_JSSHELL=1
 export CFLAGS="$SLKCFLAGS"
-export CXXFLAGS="$SLKCFLAGS"
+export CXXFLAGS="$SLKCFLAGS -fno-delete-null-pointer-checks"
 export MOZ_MAKE_FLAGS="$NUMJOBS"
 
 # Clear some variables that could break the build
@@ -283,6 +300,8 @@
 # Add the $OPTIONS above to .mozconfig:
 for option in $OPTIONS; do echo "ac_add_options $option" >> .mozconfig; done
 
+echo 'ac_add_options --enable-optimize="-g0 -02"' >> .mozconfig
+
 # Disable tests on standard build:
 echo "ac_add_options --disable-tests" >> .mozconfig
 ./mach build || exit 1
