--- a/slackpkg.SlackBuild	2021-02-08 13:52:21.581301364 +0000
+++ b/slackpkg.SlackBuild	2021-02-08 13:56:11.429378774 +0000
@@ -113,10 +113,27 @@
 mkdir -pv $PKG/var/lib/slackpkg
 mkdir -pv $PKG/var/cache/packages
 
+# support slarm64 aarch64 / riscv64
+( cd $PKG/etc/slackpkg
+  xzcat $CWD/0014-slackpkg.conf.new.patch.xz | patch -p1 --verbose || exit 1
+) || exit 1
+( cd $PKG/usr/libexec/slackpkg
+  xzcat $CWD/0010-add-support-aarch64-riscv64-core-functions.patch.xz | patch -p1 --verbose || exit 1
+) || exit 1
+( cd $PKG/usr/doc/slackpkg-$VERSION
+  xzcat $CWD/0011-mirrors-aarch64.sample.patch.xz | patch -p1 --verbose || exit 1
+  xzcat $CWD/0012-mirrors-riscv64.sample.patch.xz | patch -p1 --verbose || exit 1
+) || exit 1
+
 mkdir -p $PKG/install
 cat $CWD/doinst.sh | sed -e "s/@VERSION@/$VERSION/g" > $PKG/install/doinst.sh
 cat $CWD/slack-desc > $PKG/install/slack-desc
 
+# support slarm64 aarch64 / riscv64
+( cd $PKG/install
+  xzcat $CWD/0013-doinst.sh.patch.xz | patch -p1 --verbose || exit 1
+) || exit 1
+
 cd $PKG
 /sbin/makepkg -l y -c n $TMP/$PKGNAM-$VERSION-$ARCH-$BUILD.txz
 
