--- a/aalib.SlackBuild	2018-04-25 19:07:45.812489050 +0300
+++ b/aalib.SlackBuild	2018-04-25 19:08:38.500784723 +0300
@@ -61,6 +61,9 @@
 elif [ "$ARCH" = "x86_64" ]; then
   SLKCFLAGS="-O2 -fPIC"
   LIBDIRSUFFIX="64"
+elif [ "$ARCH" = "aarch64" ]; then
+  SLKCFLAGS="-O2"
+  LIBDIRSUFFIX="64"
 else
   SLKCFLAGS="-O2"
   LIBDIRSUFFIX=""
@@ -94,6 +97,7 @@
   --mandir=/usr/man \
   --program-prefix= \
   --program-suffix= \
+  --host=$ARCH-slackware-linux \
   --build=$ARCH-slackware-linux || exit 1
 
 # Build and install:
