--- a/ksh93.SlackBuild	2018-04-19 23:24:44.351088972 +0100
+++ b/ksh93.SlackBuild	2018-04-23 17:01:35.521103801 +0100
@@ -61,6 +61,8 @@
 elif [ "$ARCH" = "armv7hl" ]; then
   SLKCFLAGS="-O3 -march=armv7-a -mfpu=vfpv3-d16"
   LIBDIRSUFFIX=""
+elif [ "$ARCH" = "aarch64" ]; then
+  LIBDIRSUFFIX="64"
 else
   SLKCFLAGS="-O2"
   LIBDIRSUFFIX=""
@@ -85,7 +87,7 @@
   -exec chmod 644 {} \;
 
 # Configure, build, and install:
-export CC=clang
+export CC=gcc
 export CFLAGS="$SLKCFLAGS"
 mkdir build
 cd build
