--- a/usb_modeswitch.SlackBuild	2017-09-23 00:30:07.760456452 +0100
+++ b/usb_modeswitch.SlackBuild	2017-09-23 00:31:59.936456506 +0100
@@ -54,6 +54,9 @@
 elif [ "$ARCH" = "x86_64" ]; then
   SLKCFLAGS="-O2 -fPIC"
   LIBDIRSUFFIX="64"
+elif [ "$ARCH" = "aarch64" ]; then
+  SLKCFLAGS="-O2"
+  LIBDIRSUFFIX="64"
 else
   SLKCFLAGS="-O2"
   LIBDIRSUFFIX=""
@@ -72,9 +75,11 @@
  \( -perm 666 -o -perm 664 -o -perm 600 -o -perm 444 -o -perm 440 -o -perm 400 \) \
  -exec chmod 644 {} \;
 
+xzcat $CWD/add_ext_opts.diff.xz | patch -p1 --verbose || exit 1
+
 make clean || exit 1
 # "make static" uses the embedded "jim" Tcl interpreter
-make static || exit 1
+make EXT_OPTS="--build=aarch64-unknown-linux" static || exit 1
 
 make \
   DESTDIR=$PKG \
@@ -145,4 +150,3 @@
 
 cd $PKG
 /sbin/makepkg -l y -c n $TMP/$PKGNAM-$VERSION-$ARCH-$BUILD.txz
-
