--- a/alsa-lib.SlackBuild	2021-06-14 18:25:28.158003448 +0000
+++ b/alsa-lib.SlackBuild	2022-06-04 03:26:56.801833232 +0000
@@ -50,6 +50,12 @@
 elif [ "$ARCH" = "x86_64" ]; then
   SLKCFLAGS="-O2 -fPIC"
   LIBDIRSUFFIX="64"
+elif [ "$ARCH" = "aarch64" ]; then
+  SLKCFLAGS="-O2 -fPIC"
+  LIBDIRSUFFIX="64"
+elif [ "$ARCH" = "riscv64" ]; then
+  SLKCFLAGS="-O2 -fPIC"
+  LIBDIRSUFFIX="64"
 else
   SLKCFLAGS="-O2"
   LIBDIRSUFFIX=""
@@ -92,6 +98,10 @@
   done
 )
 
+xzcat $CWD/amlogic-0001-WIP-add-confs-for-gx-sound-card-and-axg-sound-card.patch.xz | patch -p1 --verbose || exit 1
+xzcat $CWD/amlogic-0002-WIP-add-Amlogic-cards-to-Makefile.am.patch.xz | patch -p1 --verbose || exit 1
+xzcat $CWD/amlogic-0003-WIP-add-analogue-surrounds-to-confs.patch.xz | patch -p1 --verbose || exit 1
+
 chown -R root:root .
 find . \
   \( -perm 777 -o -perm 775 -o -perm 711 -o -perm 555 -o -perm 511 \) \
@@ -99,6 +109,9 @@
   \( -perm 666 -o -perm 664 -o -perm 600 -o -perm 444 -o -perm 440 -o -perm 400 \) \
   -exec chmod 644 {} \+
 
+# reconfiguration after patching
+autoreconf -vif
+
 CFLAGS="$SLKCFLAGS" \
  ./configure \
   --libdir=/usr/lib${LIBDIRSUFFIX} \
