--- a/autofs.SlackBuild	2021-02-13 01:27:59.881791100 +0200
+++ b/autofs.SlackBuild	2021-02-13 01:28:32.189772994 +0200
@@ -55,6 +55,12 @@
 elif [ "$ARCH" = "x86_64" ]; then
   SLKCFLAGS="-O2 -fPIC"
   LIBDIRSUFFIX="64"
+elif [ "$ARCH" = "aarch64" ]; then
+  SLKCFLAGS="-O2 -fgnu89-inline"
+  LIBDIRSUFFIX="64"
+elif [ "$ARCH" = "riscv64" ]; then
+  SLKCFLAGS="-O2 -fgnu89-inline"
+  LIBDIRSUFFIX="64"
 else
   SLKCFLAGS="-O2"
   LIBDIRSUFFIX=""
@@ -82,6 +88,7 @@
 
 CFLAGS="$SLKCFLAGS" \
 CPPFLAGS="$SLKCFLAGS" \
+LDFLAGS="-Wl,-z,now" \
 ./configure \
   --prefix=/usr \
   --mandir=/usr/man \
@@ -91,6 +98,11 @@
   --without-sasl \
   --build=$ARCH-slackware-linux || exit 1
 
+# Nasty hack since adding -fgnu89-inline to CFLAGS is not enough.
+# https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=777787
+sed -i 's?-rdynamic?-rdynamic -fgnu89-inline?g' daemon/Makefile
+sed -i '/LIBS = -ldl/aDAEMON_LDFLAGS += -fgnu89-inline' daemon/Makefile
+
 make $NUMJOBS || make || exit 1
 make install DESTDIR=$PKG || exit 1
 
