--- a/xz.SlackBuild	2021-01-20 18:13:08.563005851 +0000
+++ b/xz.SlackBuild	2021-02-05 19:45:19.438960789 +0000
@@ -66,6 +66,12 @@
   armel)     SLKCFLAGS="-O2 -march=armv4t"
              LIBDIRSUFFIX=""
              ;;
+  aarch64)   SLKCFLAGS="-O2 -fPIC"
+             LIBDIRSUFFIX="64"
+             ;;
+  riscv64)   SLKCFLAGS="-O2 -fPIC"
+             LIBDIRSUFFIX="64"
+             ;;
   *)         SLKCFLAGS="-O2"
              LIBDIRSUFFIX=""
              ;;
