--- a/herqq.SlackBuild	2017-09-29 19:50:14.430562216 +0300
+++ b/herqq.SlackBuild	2017-09-29 20:17:06.265562284 +0300
@@ -52,6 +52,9 @@
 elif [ "$ARCH" = "armv7hl" ]; then
   SLKCFLAGS="-O2 -march=armv7-a -mfpu=vfpv3-d16"
   LIBDIRSUFFIX=""
+elif [ "$ARCH" = "aarch64" ]; then
+  SLKCFLAGS="-O2"
+  LIBDIRSUFFIX="64"
 else
   SLKCFLAGS="-O2"
   LIBDIRSUFFIX=""
@@ -68,6 +71,12 @@
 unzip $CWD/$PKGNAM-$VERSION.zip || exit 1
 cd $PKGNAM-$VERSION || exit 1
 
+xzcat $CWD/herqq-1.0.0-c++11.patch.xz | patch -p1 --verbose || exit 1
+xzcat $CWD/herqq-r129.patch.xz | patch -p1 --verbose || exit 1
+#xzcat $CWD/herqq-1.0.0-qtsoap-library.patch.xz | patch -p1 --verbose || exit 1
+xzcat $CWD/herqq-1.0.0-alt-constructor.patch.xz | patch -p1 --verbose || exit 1
+xzcat $CWD/herqq-1.0.0-alt-check-pointer.patch.xz | patch -p1 --verbose || exit 1
+
 # Fix 64-bit installation path:
 sed -i -e "s#PREFIX/lib#&${LIBDIRSUFFIX}#" $(find . -name "*.pro")
 
