--- a/floppy.SlackBuild	2017-09-22 20:44:16.755449991 +0100
+++ b/floppy.SlackBuild	2017-09-22 20:45:30.909450026 +0100
@@ -34,13 +34,14 @@
   esac
 fi
 
-if [ "$ARCH" = "x86_64" ]; then
-  SLKCFLAGS="-O -fPIC"
-  LIBDIRSUFFIX="64"
-else
-  SLKCFLAGS="-O"
-  LIBDIRSUFFIX=""
-fi
+case "$ARCH" in
+  x86_64|aarch64) SLKCFLAGS="-O -fPIC"
+                  LIBDIRSUFFIX="64"
+                  ;;
+               *) SLKCFLAGS="-O"
+                  LIBDIRSUFFIX=""
+                  ;;
+esac
 
 CWD=$(pwd)
 TMP=${TMP:-/tmp}
@@ -54,6 +55,7 @@
 cd fdutils-$VERSION
 zcat $CWD/fdutils-5.5-20081027.diff.gz | patch -p1 --verbose --backup --suffix=.orig || exit 1
 zcat $CWD/fdutils.mediaprm.diff.gz | patch -p1 --verbose --backup --suffix=.orig || exit 1
+xzcat $CWD/fdutils-alt-nodvi.patch.xz | patch -p1 --verbose --backup --suffix=.orig || exit 1
 sed -i 's|<linux/ext2_fs.h>|<ext2fs/ext2_fs.h>|' src/fdmount.c
 CFLAGS="$SLKCFLAGS" \
 ./configure \
@@ -111,4 +113,3 @@
 cat $CWD/slack-desc > $PKG/install/slack-desc
 cd $PKG
 makepkg -l y -c n $TMP/floppy-$VERSION-$ARCH-$BUILD.txz
-
