--- setup.xwmconfig.orig	2009-05-14 20:08:55.000000000 +0100
+++ setup.xwmconfig	2009-05-14 20:08:18.000000000 +0100
@@ -1,5 +1,12 @@
 #!/bin/sh
 #BLURB="Choose a default window manager for X"
+
+# These platforms are headless so we don't need to configure
+# this package.
+#
+# Silently exit if we find ourselves on one of these platforms
+egrep -q "SheevaPlug" /proc/cpuinfo > /dev/null 2>&1 && exit
+
 TMP=/var/log/setup/tmp
 if [ -r $TMP/SeTT_PX ]; then
  T_PX="`cat $TMP/SeTT_PX`"
