--- a/epic5.SlackBuild	2018-11-30 21:11:41.188329427 +0000
+++ b/epic5.SlackBuild	2018-12-01 14:17:54.180465569 +0000
@@ -57,6 +57,9 @@
 elif [ "$ARCH" = "x86_64" ]; then
   SLKCFLAGS="-O2 -fPIC"
   LIBDIRSUFFIX="64"
+elif [ "$ARCH" = "aarch64" ]; then
+  SLKCFLAGS="-O2"
+  LIBDIRSUFFIX="64"
 else
   SLKCFLAGS="-O2"
   LIBDIRSUFFIX=""
@@ -81,6 +84,9 @@
 # Support OpenSSL 1.1.x:
 zcat $CWD/openssl-1.1.patch.gz | patch -p1 --verbose || exit 1
 
+# fix global environment
+unset CPPFLAGS
+
 CFLAGS="$SLKCFLAGS" \
 ./configure \
   --prefix=/usr \
