--- a/iftop.SlackBuild	2021-02-13 01:37:23.821453513 +0200
+++ b/iftop.SlackBuild	2021-02-13 01:37:49.337437205 +0200
@@ -60,6 +60,12 @@
 elif [ "$ARCH" = "x86_64" ]; then
   SLKCFLAGS="-O2 -fPIC -fcommon"
   LIBDIRSUFFIX="64"
+elif [ "$ARCH" = "aarch64" ]; then
+  SLKCFLAGS="-O2 -fPIC -fcommon"
+  LIBDIRSUFFIX="64"
+elif [ "$ARCH" = "riscv64" ]; then
+  SLKCFLAGS="-O2 -fPIC -fcommon"
+  LIBDIRSUFFIX="64"
 else
   SLKCFLAGS="-O2 -fcommon"
   LIBDIRSUFFIX=""
