--- a/mariadb.SlackBuild	2022-02-24 00:10:00.658050957 +0000
+++ b/mariadb.SlackBuild	2022-02-24 00:14:27.444933925 +0000
@@ -28,6 +28,8 @@
 
 cd $(dirname $0) ; CWD=$(pwd)
 
+# disable mroonga storage engine (doesn't build on ARM)
+
 PKGNAM=mariadb
 VERSION=${VERSION:-$(echo ${PKGNAM}-*.tar.?z | rev | cut -f 3- -d . | cut -f 1 -d - | rev)}
 BUILD=${BUILD:-1}
@@ -81,6 +83,13 @@
 elif [ "$ARCH" = "x86_64" ]; then
   SLKCFLAGS="-O2 -fPIC"
   LIBDIRSUFFIX="64"
+elif [ "$ARCH" = "aarch64" ]; then
+  SLKCFLAGS="-O2 -fPIC"
+  LIBDIRSUFFIX="64"
+elif [ "$ARCH" = "riscv64" ]; then
+  SLKCFLAGS="-O2 -fPIC"
+  SLKLDFLAGS="-latomic"
+  LIBDIRSUFFIX="64"
 fi
 
 rm -rf $PKG
@@ -99,6 +108,7 @@
 
 mkdir build
 cd build
+LDFLAGS="$SLKLDFLAGS" \
 cmake \
   -DCMAKE_C_FLAGS="$SLKCFLAGS" \
   -DCMAKE_CXX_FLAGS="$SLKCFLAGS" \
@@ -134,6 +144,7 @@
   -DCONC_WITH_SSL=ON \
   -DUSE_ARIA_FOR_TMP_TABLES=ON \
   -DAWS_SDK_EXTERNAL_PROJECT=OFF \
+  -DWITHOUT_MROONGA_STORAGE_ENGINE=1 \
   .. || exit 1
 
 # doesn't currently compile
