--- cdrtools-2.01/include/mconfig.h.orig	2007-05-05 00:40:40.000000000 +0200
+++ cdrtools-2.01/include/mconfig.h	2007-05-05 00:40:48.000000000 +0200
@@ -77,7 +77,7 @@
 #define SIZEOF_CHAR_P 8
 #define SIZEOF_UNSIGNED_LONG_INT 8
 #define SIZEOF_UNSIGNED_CHAR_P 8
-#elif defined(__i386__) || defined(__powerpc__) || defined(__s390__)
+#elif defined(__i386__) || defined(__powerpc__) || defined(__s390__) || defined(__arm__)
 #define SIZEOF_LONG_INT 4
 #define SIZEOF_CHAR_P 4
 #define SIZEOF_UNSIGNED_LONG_INT 4
