--- a/freeglut.SlackBuild	2021-02-13 00:03:39.097712358 +0200
+++ b/freeglut.SlackBuild	2021-02-13 00:04:01.005710181 +0200
@@ -61,6 +61,12 @@
 elif [ "$ARCH" = "x86_64" ]; then
   SLKCFLAGS="-O2 -fPIC -fcommon"
   LIBDIRSUFFIX="64"
+elif [ "$ARCH" = "aarch64" ]; then
+  SLKCFLAGS="-O2 -fPIC -fcommon"
+  LIBDIRSUFFIX="64"
+elif [ "$ARCH" = "riscv64" ]; then
+  SLKCFLAGS="-O2 -fPIC -fcommon"
+  LIBDIRSUFFIX="64"
 else
   SLKCFLAGS="-O2 -fcommon"
   LIBDIRSUFFIX=""
