--- a/xine-lib.SlackBuild	2021-02-13 00:44:52.792069419 +0200
+++ b/xine-lib.SlackBuild	2021-02-13 00:45:25.172035722 +0200
@@ -55,6 +55,12 @@
   LIBDIRSUFFIX=""
   # 32bit x86 fails without this... let me know if you have a fix
   FFARCHOPTS="--disable-asm"
+elif [ "$ARCH" = "aarch64" ]; then
+  LIBDIRSUFFIX="64"
+  FFARCHOPTS="--disable-asm"
+elif [ "$ARCH" = "riscv64" ]; then
+  LIBDIRSUFFIX="64"
+  FFARCHOPTS="--disable-asm"
 else
   LIBDIRSUFFIX=""
   FFARCHOPTS=""
