--- a/mesa.SlackBuild	2022-08-09 02:58:15.107303823 +0000
+++ b/mesa.SlackBuild	2022-08-10 20:09:24.804524369 +0000
@@ -33,7 +33,7 @@
 NUMJOBS=${NUMJOBS:-" -j$(expr $(nproc) + 1) "}
 
 # Be sure this list is up-to-date:
-GALLIUM_DRIVERS="nouveau,r300,r600,svga,radeonsi,swrast,virgl,iris,crocus,zink"
+GALLIUM_DRIVERS="freedreno,nouveau,swrast,virgl,kmsro,lima,panfrost,v3d,vc4"
 
 if [ -z "$ARCH" ]; then
   case "$( uname -m )" in
@@ -69,6 +69,12 @@
 elif [ "$ARCH" = "x86_64" ]; then
   SLKCFLAGS="-O2 -fPIC"
   LIBDIRSUFFIX="64"
+elif [ "$ARCH" = "aarch64" ]; then
+  SLKCFLAGS="-O2 -fPIC"
+  LIBDIRSUFFIX="64"
+elif [ "$ARCH" = "riscv64" ]; then
+  SLKCFLAGS="-O2 -fPIC"
+  LIBDIRSUFFIX="64"
 else
   SLKCFLAGS="-O2"
   LIBDIRSUFFIX=""
@@ -132,7 +138,7 @@
   -Dgallium-nine=true \
   -Dosmesa=true \
   -Dgallium-drivers=$GALLIUM_DRIVERS \
-  -Dvulkan-drivers=amd,intel,swrast \
+  -Dvulkan-drivers=amd,swrast,broadcom \
   -Dvulkan-layers=device-select,intel-nullhw,overlay \
   -Dglvnd=true \
   -Dllvm=enabled \
@@ -143,6 +149,9 @@
   -Dgles2=enabled \
   -Dopengl=true \
   -Dglx=dri \
+  -Ddri3=enabled \
+  -Dgbm=enabled \
+  -Dlibunwind=disabled \
   .. || exit 1
   "${NINJA:=ninja}" $NUMJOBS || exit 1
   DESTDIR=$PKG $NINJA install || exit 1
