--- a/mesa.SlackBuild	2022-08-10 23:39:34.530667285 +0300
+++ b/mesa.SlackBuild	2022-08-10 23:46:21.963021585 +0300
@@ -33,7 +33,7 @@
 NUMJOBS=${NUMJOBS:-" -j$(expr $(nproc) + 1) "}
 
 # Be sure this list is up-to-date:
-GALLIUM_DRIVERS="nouveau,r300,r600,svga,radeonsi,swrast,virgl,iris,crocus,zink"
+GALLIUM_DRIVERS="nouveau,r300,r600,radeonsi,swrast,virgl,zink,freedreno,etnaviv,kmsro,lima,panfrost,v3d,vc4"
 
 if [ -z "$ARCH" ]; then
   case "$( uname -m )" in
@@ -69,6 +69,12 @@
 elif [ "$ARCH" = "x86_64" ]; then
   SLKCFLAGS="-O2 -fPIC"
   LIBDIRSUFFIX="64"
+elif [ "$ARCH" = "aarch64" ]; then
+  SLKCFLAGS="-O2 -fPIC"
+  LIBDIRSUFFIX="64"
+elif [ "$ARCH" = "riscv64" ]; then
+  SLKCFLAGS="-O2 -fPIC"
+  LIBDIRSUFFIX="64"
 else
   SLKCFLAGS="-O2"
   LIBDIRSUFFIX=""
@@ -132,8 +138,8 @@
   -Dgallium-nine=true \
   -Dosmesa=true \
   -Dgallium-drivers=$GALLIUM_DRIVERS \
-  -Dvulkan-drivers=amd,intel,swrast \
-  -Dvulkan-layers=device-select,intel-nullhw,overlay \
+  -Dvulkan-drivers=amd,swrast,broadcom,panfrost \
+  -Dvulkan-layers=device-select,overlay \
   -Dglvnd=true \
   -Dllvm=enabled \
   -Dshared-llvm=enabled \
@@ -143,6 +149,8 @@
   -Dgles2=enabled \
   -Dopengl=true \
   -Dglx=dri \
+  -Ddri3=enabled \
+  -Dgbm=enabled \
   .. || exit 1
   "${NINJA:=ninja}" $NUMJOBS || exit 1
   DESTDIR=$PKG $NINJA install || exit 1
