--- a/mesa.SlackBuild	2021-02-23 20:45:49.393515112 +0200
+++ b/mesa.SlackBuild	2021-02-23 20:47:35.844867153 +0200
@@ -31,8 +31,8 @@
 NUMJOBS=${NUMJOBS:-" -j$(expr $(nproc) + 1) "}
 
 # Be sure this list is up-to-date:
-DRI_DRIVERS="i915,i965,r100,r200,nouveau"
-GALLIUM_DRIVERS="nouveau,r300,r600,svga,radeonsi,swrast,virgl,iris"
+DRI_DRIVERS="nouveau"
+GALLIUM_DRIVERS="freedreno,nouveau,swrast,virgl,kmsro,lima,panfrost,v3d,vc4"
 
 if [ -z "$ARCH" ]; then
   case "$( uname -m )" in
@@ -67,6 +67,12 @@
 elif [ "$ARCH" = "x86_64" ]; then
   SLKCFLAGS="-O2 -fPIC"
   LIBDIRSUFFIX="64"
+elif [ "$ARCH" = "aarch64" ]; then
+  SLKCFLAGS="-O2 -fPIC"
+  LIBDIRSUFFIX="64"
+elif [ "$ARCH" = "riscv64" ]; then
+  SLKCFLAGS="-O2 -fPIC"
+  LIBDIRSUFFIX="64"
 else
   SLKCFLAGS="-O2"
   LIBDIRSUFFIX=""
@@ -101,6 +107,8 @@
   done
 fi
 
+xzcat $CWD/meson.build.patch.xz | patch -p1 --verbose || exit 1
+
 # Configure, build, and install:
 export CFLAGS="$SLKCFLAGS"
 export CXXFLAGS="$SLKCFLAGS"
@@ -134,6 +142,9 @@
   -Dgles2=enabled \
   -Dopengl=true \
   -Dglx=dri \
+  -Ddri3=true \
+  -Dgbm=true \
+  -Dlibunwind=false \
   .. || exit 1
   "${NINJA:=ninja}" $NUMJOBS || exit 1
   DESTDIR=$PKG $NINJA install || exit 1
@@ -167,6 +178,7 @@
     CFLAGS="$SLKCFLAGS" \
     ./configure \
       --prefix=/usr \
+      --disable-gles1 \
       --build=$ARCH-slackware-linux || exit 1
     make $NUMJOBS || exit 1
     # Install all the demos (including the pointless ones) at first, in a
@@ -174,7 +186,8 @@
     make install DESTDIR=$PKG/cruft || exit 1
     # Install gears and glinfo, as well as a few other demos:
     mkdir -p $PKG/usr/bin
-    for demo in gears glinfo glthreads glxcontexts glxdemo glxgears \
+    for demo in eglinfo es2_info es2gears_x11 \
+      gears glinfo glthreads glxcontexts glxdemo glxgears \
       glxgears_fbconfig glxheads glxinfo glxpbdemo glxpixmap ; do
       mv --verbose $PKG/cruft/usr/bin/$demo $PKG/usr/bin
   done
@@ -209,4 +222,3 @@
 
 cd $PKG
 /sbin/makepkg -l y -c n $TMP/${PKGNAM}-$VERSION-$ARCH-$BUILD.txz
-
