--- a/mesa.SlackBuild	2019-10-09 22:22:08.128790109 +0300
+++ b/mesa.SlackBuild	2020-02-01 14:57:18.895492463 +0200
@@ -31,8 +31,8 @@
 NUMJOBS=${NUMJOBS:-" -j$(expr $(nproc) + 1) "}
 
 # Be sure this list is up-to-date:
-DRI_DRIVERS="i915,i965,r100,r200,nouveau"
-GALLIUM_DRIVERS="nouveau,r300,r600,svga,radeonsi,swrast,virgl,iris"
+DRI_DRIVERS="nouveau"
+GALLIUM_DRIVERS="freedreno,nouveau,swrast,virgl,kmsro,lima,panfrost,v3d,vc4"
 
 if [ -z "$ARCH" ]; then
   case "$( uname -m )" in
@@ -67,6 +67,9 @@
 elif [ "$ARCH" = "x86_64" ]; then
   SLKCFLAGS="-O2 -fPIC"
   LIBDIRSUFFIX="64"
+elif [ "$ARCH" = "aarch64" ]; then
+  SLKCFLAGS="-O2"
+  LIBDIRSUFFIX="64"
 else
   SLKCFLAGS="-O2"
   LIBDIRSUFFIX=""
@@ -129,9 +132,13 @@
   -Dllvm=true \
   -Dshared-llvm=true \
   -Dshared-glapi=true \
+  -Ddri3=true \
   -Degl=true \
-  -Dgles1=true \
+  -Dgbm=true \
+  -Dgles1=false \
   -Dgles2=true \
+  -Dglx=dri \
+  -Dlibunwind=false \
   .. || exit 1
   "${NINJA:=ninja}" $NUMJOBS || exit 1
   DESTDIR=$PKG $NINJA install || exit 1
@@ -158,6 +165,7 @@
     CFLAGS="$SLKCFLAGS" \
     ./configure \
       --prefix=/usr \
+      --disable-gles1 \
       --build=$ARCH-slackware-linux || exit 1
     make $NUMJOBS || exit 1
     # Install all the demos (including the pointless ones) at first, in a
@@ -165,7 +173,8 @@
     make install DESTDIR=$PKG/cruft || exit 1
     # Install gears and glinfo, as well as a few other demos:
     mkdir -p $PKG/usr/bin
-    for demo in gears glinfo glthreads glxcontexts glxdemo glxgears \
+    for demo in eglinfo es2_info es2gears_x11 \
+      gears glinfo glthreads glxcontexts glxdemo glxgears \
       glxgears_fbconfig glxheads glxinfo glxpbdemo glxpixmap ; do
       mv --verbose $PKG/cruft/usr/bin/$demo $PKG/usr/bin
   done
