--- a/mesa.SlackBuild	2019-10-09 20:22:08.128790109 +0100
+++ b/mesa.SlackBuild	2019-10-10 20:01:17.195858568 +0100
@@ -31,8 +31,8 @@
 NUMJOBS=${NUMJOBS:-" -j$(expr $(nproc) + 1) "}
 
 # Be sure this list is up-to-date:
-DRI_DRIVERS="i915,i965,r100,r200,nouveau"
-GALLIUM_DRIVERS="nouveau,r300,r600,svga,radeonsi,swrast,virgl,iris"
+DRI_DRIVERS="nouveau"
+GALLIUM_DRIVERS="freedreno,nouveau,swrast,virgl,kmsro,lima,panfrost,v3d,vc4"
 
 if [ -z "$ARCH" ]; then
   case "$( uname -m )" in
@@ -67,6 +67,9 @@
 elif [ "$ARCH" = "x86_64" ]; then
   SLKCFLAGS="-O2 -fPIC"
   LIBDIRSUFFIX="64"
+elif [ "$ARCH" = "aarch64" ]; then
+  SLKCFLAGS="-O2"
+  LIBDIRSUFFIX="64"
 else
   SLKCFLAGS="-O2"
   LIBDIRSUFFIX=""
@@ -158,6 +161,7 @@
     CFLAGS="$SLKCFLAGS" \
     ./configure \
       --prefix=/usr \
+      --disable-gles1 \
       --build=$ARCH-slackware-linux || exit 1
     make $NUMJOBS || exit 1
     # Install all the demos (including the pointless ones) at first, in a
