--- a/linuxdoc-tools.build	2017-11-18 01:05:11.076547593 +0200
+++ b/linuxdoc-tools.build	2017-11-18 01:09:46.713547593 +0200
@@ -102,7 +102,8 @@
   --prefix=/usr \
   --sysconfdir=/etc \
   --docdir=/usr/doc/asciidoc-$ASCIIDOCVER \
-  --mandir=/usr/man || exit 1
+  --mandir=/usr/man \
+  --build=$ARCH-slackware-linux$HOSTTARGET || exit 1
 
 # Install:
 make install && make docs || exit 1
@@ -326,6 +327,7 @@
   --enable-http \
   --enable-default-catalog=/etc/sgml/catalog \
   --enable-default-search-path=/usr/share/sgml \
+  --build=$ARCH-slackware-linux$HOSTTARGET \
   || exit 1
 
 # Build:
@@ -406,6 +408,7 @@
   --enable-default-catalog=/etc/sgml/catalog \
   --enable-default-search-path=/usr/share/sgml \
   --datadir=/usr/share/sgml/openjade-$OPENJADEVER \
+  --build=$ARCH-slackware-linux$HOSTTARGET \
   || exit 1
 
 # Build:
