--- a/libtheora.SlackBuild	2017-09-27 19:38:26.020408546 +0300
+++ b/libtheora.SlackBuild	2017-09-27 19:39:44.611408546 +0300
@@ -52,6 +52,9 @@
 elif [ "$ARCH" = "x86_64" ]; then
   SLKCFLAGS="-O2 -fPIC"
   LIBDIRSUFFIX="64"
+elif [ "$ARCH" = "aarch64" ]; then
+  SLKCFLAGS="-O2"
+  LIBDIRSUFFIX="64"
 else
   SLKCFLAGS="-O2"
   LIBDIRSUFFIX=""
@@ -70,6 +73,8 @@
   \( -perm 666 -o -perm 664 -o -perm 600 -o -perm 444 -o -perm 440 -o -perm 400 \) \
   -exec chmod 644 {} \;
 
+xzcat $CWD/libtheora-1.1.1-libpng16.patch.xz | patch -p0 --verbose || exit 1
+
 # Configure:
 CFLAGS="$SLKCFLAGS" \
 ./configure \
@@ -129,4 +134,3 @@
 
 cd $TMP/package-${PKGNAM}
 /sbin/makepkg -l y -c n $TMP/${PKGNAM}-$VERSION-$ARCH-$BUILD.txz
-
