--- a/aalib.SlackBuild	2021-02-13 03:06:27.064000316 +0200
+++ b/aalib.SlackBuild	2021-02-13 03:07:00.403983273 +0200
@@ -61,6 +61,12 @@
 elif [ "$ARCH" = "x86_64" ]; then
   SLKCFLAGS="-O2 -fPIC"
   LIBDIRSUFFIX="64"
+elif [ "$ARCH" = "aarch64" ]; then
+  SLKCFLAGS="-O2 -fPIC"
+  LIBDIRSUFFIX="64"
+elif [ "$ARCH" = "riscv64" ]; then
+  SLKCFLAGS="-O2 -fPIC"
+  LIBDIRSUFFIX="64"
 else
   SLKCFLAGS="-O2"
   LIBDIRSUFFIX=""
@@ -94,6 +100,7 @@
   --mandir=/usr/man \
   --program-prefix= \
   --program-suffix= \
+  --host=$ARCH-slackware-linux \
   --build=$ARCH-slackware-linux || exit 1
 
 # Build and install:
