--- a/procmail.SlackBuild	2017-10-01 05:58:48.345864257 +0300
+++ b/procmail.SlackBuild	2017-10-01 06:00:16.019864257 +0300
@@ -49,6 +49,7 @@
 cd procmail-$VERSION || exit 1
 zcat $CWD/procmail_3.22-5.diff.gz | patch -p1 --verbose || exit 1
 zcat $CWD/procmail.lfs.diff.gz | patch -p1 --verbose || exit 1
+xzcat $CWD/procmail-3.22-getline.patch.xz | patch -p1 --verbose || exit 1
 chown -R root:root .
 find . \
   \( -perm 777 -o -perm 775 -o -perm 711 -o -perm 555 -o -perm 511 \) \
