diff -urN radeontool-1.5.orig/radeontool.c radeontool-1.5/radeontool.c
--- radeontool-1.5.orig/radeontool.c	2004-02-11 04:50:27.000000000 +0100
+++ radeontool-1.5/radeontool.c	2006-09-18 09:43:38.000000000 +0200
@@ -21,10 +21,11 @@
 #include <sys/stat.h>
 #include <fcntl.h>
 #include <sys/mman.h>
-#include <asm/page.h>
 
 #include "radeon_reg.h"
 
+#define PAGE_SIZE sysconf(_SC_PAGE_SIZE)
+
 int debug;
 int skip;

