--- a/mozilla-thunderbird.SlackBuild	2018-08-11 01:31:29.100615723 +0100
+++ b/mozilla-thunderbird.SlackBuild	2018-08-13 18:08:03.305484133 +0100
@@ -87,6 +87,10 @@
   SLKCFLAGS="-march=armv4 -mtune=xscale"
   LIBDIRSUFFIX=""
   OPTIMIZE=${OPTIMIZE:-"-O1"}
+elif [ "$ARCH" = "aarch64" ]; then
+  SLKCFLAGS=""
+  LIBDIRSUFFIX="64"
+  OPTIMIZE=${OPTIMIZE:-"-O1"}
 else
   SLKCFLAGS=""
   LIBDIRSUFFIX=""
@@ -194,6 +198,15 @@
 
 # Our building options, in a configure-like display ;)
 OPTIONS="\
+  --enable-extensions=default \
+  --enable-system-ffi \
+  --enable-pie \
+  --disable-eme \
+  --disable-gconf \
+  --with-system-libvpx \
+  --disable-stylo \
+  --disable-webrtc \
+  \
   --enable-official-branding \
   --prefix=/usr \
   --libdir=/usr/lib${LIBDIRSUFFIX} \
@@ -228,6 +241,10 @@
     mail/installer/package-manifest.in || exit 1
 fi
 
+export SLKCFLAGS="$SLKCFLAGS -g -fstack-protector --param=ssp-buffer-size=4 -Wall -O2 -fno-schedule-insns"
+export SLKCXXFLAGS="$SLKCFLAGS"
+export LDFLAGS="-Wl,--no-as-needed -Wl,--no-keep-memory"
+
 export BUILD_OFFICIAL=1
 export MOZILLA_OFFICIAL=1
 export MOZ_PHOENIX=1
