--- a/mozilla-thunderbird.SlackBuild	2019-02-01 06:39:48.930204777 +0200
+++ b/mozilla-thunderbird.SlackBuild	2019-02-01 06:42:44.216665806 +0200
@@ -87,6 +87,10 @@
   SLKCFLAGS="-march=armv4 -mtune=xscale"
   LIBDIRSUFFIX=""
   OPTIMIZE=${OPTIMIZE:-"-O1"}
+elif [ "$ARCH" = "aarch64" ]; then
+  SLKCFLAGS=""
+  LIBDIRSUFFIX="64"
+  OPTIMIZE=${OPTIMIZE:-"-O1"}
 else
   SLKCFLAGS=""
   LIBDIRSUFFIX=""
@@ -111,6 +115,12 @@
   SLKLDFLAGS=" -Wl,--as-needed -Wl,--reduce-memory-overheads -Wl,--no-keep-memory -Wl,--stats"
   export LDFLAGS="$SLKLDFLAGS"
   export MOZ_LINK_FLAGS="$SLKLDFLAGS"
+elif [ "$ARCH" = "aarch64" ]; then
+  # This might also help with the linker memory situation on some $ARCH. Feel free
+  # to match any other $ARCH that could benefit from this.
+  SLKLDFLAGS=" -Wl,--as-needed -Wl,--reduce-memory-overheads -Wl,--no-keep-memory -Wl,--stats"
+  export LDFLAGS="$SLKLDFLAGS"
+  export MOZ_LINK_FLAGS="$SLKLDFLAGS"
 else
   # Link using gold.  This also helps prevent running out of memory on 32-bit
   # systems, and avoids a recurring build failure with GNU ld on other systems.
@@ -176,6 +186,9 @@
              # Make firefox compile on ARM platforms lacking neon support:
              zcat $CWD/mozilla-firefox.xpcom_arm.patch.gz | patch -p1 --verbose || exit 1
              ;;
+    aarch64) ARCH_CONFIG="--disable-elf-hack"
+             xzcat $CWD/Bug-1238661---fix-mozillaSignalTrampoline-to-work-.patch.xz | patch -p1 --verbose || exit 1
+             ;;
     *)       ARCH_CONFIG=" "
              ;;
 esac
@@ -260,6 +273,12 @@
 # Add the $OPTIONS above to .mozconfig:
 for option in $OPTIONS; do echo "ac_add_options $option" >> .mozconfig; done
 
+# ARM: see https://bugzilla.redhat.com/show_bug.cgi?id=1426850
+# This isn't added using $SLKCONFARGS because it doesn't cater for things with spaces in:
+# unless I edit the for loop above to use IFS with some other variable.
+# this is easier right now ;-)
+echo 'ac_add_options --enable-optimize="${OPTIMIZE} -fno-schedule-insns"' >> .mozconfig
+
 # Disable tests on standard build:
 echo "ac_add_options --disable-tests" >> .mozconfig
 ./mach build || exit 1
