--- a/speexdsp.SlackBuild	2021-02-13 03:51:55.534478648 +0200
+++ b/speexdsp.SlackBuild	2021-02-13 03:52:29.458455897 +0200
@@ -62,6 +62,12 @@
 elif [ "$ARCH" = "x86_64" ]; then
   SLKCFLAGS="-O2 -fPIC"
   LIBDIRSUFFIX="64"
+elif [ "$ARCH" = "aarch64" ]; then
+  SLKCFLAGS="-O2 -fPIC"
+  LIBDIRSUFFIX="64"
+elif [ "$ARCH" = "riscv64" ]; then
+  SLKCFLAGS="-O2 -fPIC"
+  LIBDIRSUFFIX="64"
 else
   SLKCFLAGS="-O2"
   LIBDIRSUFFIX=""
@@ -88,6 +94,7 @@
   --mandir=/usr/man \
   --docdir=/usr/doc/$PKGNAM-$VERSION \
   --enable-static=no \
+  --disable-neon \
   --build=$ARCH-slackware-linux || exit 1
 
 make $NUMJOBS || make || exit 1
