--- a/yasm.SlackBuild	2017-10-14 07:12:27.160557770 +0300
+++ b/yasm.SlackBuild	2017-10-14 07:13:05.451557770 +0300
@@ -96,6 +96,9 @@
   x86_64)    SLKCFLAGS="-O2 -fPIC"
              SLKLDFLAGS="-L/usr/lib64"; LIBDIRSUFFIX="64"
              ;;
+ aarch64)    SLKCFLAGS="-O2"
+             SLKLDFLAGS="-L/usr/lib64"; LIBDIRSUFFIX="64"
+             ;;
   athlon-xp) SLKCFLAGS="-march=athlon-xp -O3 -pipe -fomit-frame-pointer"
              SLKLDFLAGS=""; LIBDIRSUFFIX=""
              ;;
