--- a/gv.SlackBuild	2017-10-04 20:07:46.976270038 +0300
+++ b/gv.SlackBuild	2017-10-04 20:09:18.497270038 +0300
@@ -49,6 +49,8 @@
   SLKCFLAGS="-O2"
 elif [ "$ARCH" = "x86_64" ]; then
   SLKCFLAGS="-O2 -fPIC"
+elif [ "$ARCH" = "aarch64" ]; then
+  SLKCFLAGS="-O2"
 else
   SLKCFLAGS="-O2"
 fi
@@ -58,7 +60,8 @@
 tar xvf $CWD/${PKGNAM}-$VERSION.tar.?z* || exit 1
 cd ${PKGNAM}-$VERSION || exit 1
 
-zcat $CWD/gv-xft.patch.gz | patch -p1 --verbose || exit 1
+#zcat $CWD/gv-xft.patch.gz | patch -p1 --verbose || exit 1
+xzcat $CWD/gv-aarch64.patch.xz | patch -p1 --verbose || exit 1
 
 # Make sure ownerships and permissions are sane:
 chown -R root:root .
