--- a/sdl.SlackBuild	2021-02-13 03:50:17.542544517 +0200
+++ b/sdl.SlackBuild	2021-02-13 03:50:56.350518404 +0200
@@ -61,6 +61,12 @@
 elif [ "$ARCH" = "x86_64" ]; then
   SLKCFLAGS="-O2 -fPIC"
   LIBDIRSUFFIX="64"
+elif [ "$ARCH" = "aarch64" ]; then
+  SLKCFLAGS="-O2 -fPIC"
+  LIBDIRSUFFIX="64"
+elif [ "$ARCH" = "riscv64" ]; then
+  SLKCFLAGS="-O2 -fPIC"
+  LIBDIRSUFFIX="64"
 else
   SLKCFLAGS="-O2"
   LIBDIRSUFFIX=""
@@ -97,7 +103,9 @@
   --disable-esd \
   --enable-shared=yes \
   --enable-static=no \
-  --disable-x11-shared || exit 1
+  --disable-x11-shared \
+  --host=$ARCH-slackware-linux \
+  --build=$ARCH-slackware-linux || exit 1
 
 make $NUMJOBS || make || exit 1
 
@@ -139,7 +147,9 @@
   --enable-static=no \
   --enable-jpg-shared=no \
   --enable-png-shared=no \
-  --enable-tif-shared=no || exit 1
+  --enable-tif-shared=no \
+  --host=$ARCH-slackware-linux \
+  --build=$ARCH-slackware-linux || exit 1
 
 make $NUMJOBS || make || exit 1
 make install DESTDIR=$PKG || exit 1
@@ -166,7 +176,9 @@
   --libdir=/usr/local/lib${LIBDIRSUFFIX} \
   --with-pic \
   --enable-shared=no \
-  --enable-static=yes || exit 1
+  --enable-static=yes \
+  --host=$ARCH-slackware-linux \
+  --build=$ARCH-slackware-linux || exit 1
   make $NUMJOBS || make || exit 1
   make install || exit 1
 ) || exit 1
@@ -185,7 +197,9 @@
   --mandir=/usr/man \
   --enable-music-mod \
   --enable-shared=yes \
-  --enable-static=no || exit 1
+  --enable-static=no \
+  --host=$ARCH-slackware-linux \
+  --build=$ARCH-slackware-linux || exit 1
 
 make $NUMJOBS || make || exit 1
 make install DESTDIR=$PKG || exit 1
@@ -216,7 +230,9 @@
   --libdir=/usr/lib${LIBDIRSUFFIX} \
   --mandir=/usr/man \
   --enable-shared=yes \
-  --enable-static=no || exit 1
+  --enable-static=no \
+  --host=$ARCH-slackware-linux \
+  --build=$ARCH-slackware-linux || exit 1
 
 make $NUMJOBS || make || exit 1
 make install DESTDIR=$PKG || exit 1
@@ -247,7 +263,9 @@
   --libdir=/usr/lib${LIBDIRSUFFIX} \
   --mandir=/usr/man \
   --enable-shared=yes \
-  --enable-static=no || exit 1
+  --enable-static=no \
+  --host=$ARCH-slackware-linux \
+  --build=$ARCH-slackware-linux || exit 1
 
 make $NUMJOBS || make || exit 1
 make install DESTDIR=$PKG || exit 1
@@ -277,7 +295,9 @@
   --sysconfdir=/etc \
   --mandir=/usr/man \
   --enable-shared=yes \
-  --enable-static=no || exit 1
+  --enable-static=no \
+  --host=$ARCH-slackware-linux \
+  --build=$ARCH-slackware-linux || exit 1
 
 make $NUMJOBS || make || exit 1
 make install DESTDIR=$PKG || exit 1
