--- a/perl.SlackBuild	2021-02-10 21:00:07.172197428 +0200
+++ b/perl.SlackBuild	2021-02-10 21:01:28.832130185 +0200
@@ -108,6 +108,12 @@
 elif [ "$ARCH" = "x86_64" ]; then
   SLKCFLAGS="-O2 -fPIC"
   LIBDIRSUFFIX="64"
+elif [ "$ARCH" = "aarch64" ]; then
+  SLKCFLAGS="-O2 -fPIC"
+  LIBDIRSUFFIX="64"
+elif [ "$ARCH" = "riscv64" ]; then
+  SLKCFLAGS="-O2 -fPIC"
+  LIBDIRSUFFIX="64"
 fi
 
 if [ -x /usr/bin/perl ]; then
@@ -146,7 +152,7 @@
 #
 USE_THREADS="-Dusethreads -Duseithreads"
 
-if [ "$ARCH" = "x86_64" ]; then # adopted from "Cross Linux From Scratch"
+if [ "$ARCH" = "x86_64" -o "$ARCH" = "aarch64" -o "$ARCH" = "riscv64" ]; then # adopted from "Cross Linux From Scratch"
   # Configure must be told to also use lib64:
   zcat $CWD/perl.configure.multilib.patch.gz | patch -p1 --verbose || exit 1
   # "perl -V" should report that libc is in /lib64
