--- a/openssl10.SlackBuild	2018-05-08 22:05:33.339257414 +0300
+++ b/openssl10.SlackBuild	2018-05-10 18:41:07.815009375 +0300
@@ -86,6 +86,8 @@
   LIBDIRSUFFIX=""
 elif [ "$ARCH" = "x86_64" ]; then
   LIBDIRSUFFIX="64"
+elif [ "$ARCH" = "aarch64" ]; then
+  LIBDIRSUFFIX="64"
 fi
 
 # OpenSSL has a (nasty?) habit of bumping the internal version number with
@@ -137,6 +139,8 @@
  no-weak-ssl-ciphers \
  shared
 
+sed -i "s/^LIBDIR=lib$/LIBDIR=lib${LIBDIRSUFFIX}/" Makefile
+
 make $NUMJOBS depend || make depend || exit 1
 
 make $NUMJOBS || make || exit 1
