--- a/rust.SlackBuild	2021-05-22 20:56:48.085331336 +0300
+++ b/rust.SlackBuild	2021-05-22 21:06:58.986056968 +0300
@@ -98,6 +98,9 @@
 elif [ "$ARCH" = "armv7hl" ]; then
   BARCH="armv7"
   TARCH="$BARCH"
+elif [ "$ARCH" = "riscv64" ]; then
+  BARCH="riscv64gc"
+  TARCH="$BARCH"
 else
   BARCH="$ARCH"
   TARCH="$ARCH"
@@ -126,6 +129,10 @@
   LIBDIRSUFFIX="64"
 elif [ "$ARCH" = "armv7hl" ]; then
   LIBDIRSUFFIX=""
+elif [ "$ARCH" = "aarch64" ]; then
+  LIBDIRSUFFIX="64"
+elif [ "$ARCH" = "riscv64" ]; then
+  LIBDIRSUFFIX="64"
 else
   LIBDIRSUFFIX=""
 fi
@@ -198,6 +205,12 @@
 
 [target.armv7-unknown-linux-gnueabihf]
 llvm-config = "/usr/bin/llvm-config"
+
+[target.aarch64-unknown-linux-gnu]
+llvm-config = "/usr/bin/llvm-config"
+
+[target.riscv64gc-unknown-linux-gnu]
+llvm-config = "/usr/bin/llvm-config"
 EOF
 fi
 
@@ -205,6 +218,11 @@
   sed -i "s|^\(extended = true\)$|\1\nrustc = \"/usr/bin/rustc\"\ncargo = \"/usr/bin/cargo\"|" config.toml
 fi
 
+if [ "$ARCH" = "riscv64" ] ; then
+  # "analysis" does not compile for riscv64 architecture
+  sed -i "s|^\(extended = true\)$|\1\ntools = \[\"cargo\", \"rls\", \"clippy\", \"rustfmt\", \"src\"\]|" config.toml
+fi
+
 chown -R root:root .
 find -L . \
  \( -perm 777 -o -perm 775 -o -perm 750 -o -perm 711 -o -perm 555 \
