--- a/rust.SlackBuild	2020-01-30 18:55:20.108351291 +0000
+++ b/rust.SlackBuild	2020-01-31 22:39:54.223714399 +0000
@@ -124,6 +124,9 @@
   LIBDIRSUFFIX="64"
 elif [ "$ARCH" = "armv7hl" ]; then
   LIBDIRSUFFIX=""
+elif [ "$ARCH" = "aarch64" ]; then
+  SLKCFLAGS="-O2"
+  LIBDIRSUFFIX="64"
 else
   LIBDIRSUFFIX=""
 fi
@@ -196,6 +199,9 @@
 
 [target.armv7-unknown-linux-gnueabihf]
 llvm-config = "/usr/bin/llvm-config"
+
+[target.aarch64-unknown-linux-gnu]
+llvm-config = "/usr/bin/llvm-config"
 EOF
 fi
 
