--- a/qtwebkit/qtwebkit.SlackBuild	2019-09-29 23:48:32.140502596 +0000
+++ b/qtwebkit/qtwebkit.SlackBuild	2020-11-02 22:40:14.026609467 +0000
@@ -66,6 +66,10 @@
   SLKCFLAGS="-O2 -march=armv7-a -mfpu=vfpv3-d16 -fno-strict-volatile-bitfields -std=gnu++98 -fpermissive -fno-delete-null-pointer-checks"
   LIBDIRSUFFIX=""
   WEBKITFLAGS="--no-force-sse2"
+elif [ "$ARCH" = "aarch64" ]; then
+  SLKCFLAGS="-O2 -fstack-protector-strong -Wformat -Werror=format-security -fno-strict-volatile-bitfields -D_FORTIFY_SOURCE=2"
+  LIBDIRSUFFIX="64"
+  WEBKITFLAGS="--no-force-sse2"
 else
   SLKCFLAGS="-O2 -std=gnu++98 -fpermissive -fno-delete-null-pointer-checks"
   LIBDIRSUFFIX=""
@@ -90,6 +94,8 @@
 # Patch to fix compiling with gcc5:
 zcat $CWD/qtwebkit.putByIndexBeyondVectorLengthWithArrayStorage.diff.gz | patch -p1 --verbose || exit 1
 
+cat $CWD/fixed-generate-header-bison3.7.patch | patch -p1 --verbose || exit 1
+
 chown -R root:root .
 find . \
  \( -perm 777 -o -perm 775 -o -perm 711 -o -perm 555 -o -perm 511 \) \
@@ -97,14 +103,22 @@
  \( -perm 666 -o -perm 664 -o -perm 600 -o -perm 444 -o -perm 440 -o -perm 400 \) \
  -exec chmod 644 {} \+
 
+# Configure & build:
+# For ARM we added:
+# QMAKE_LFLAGS_RELEASE=\"$SLKLDFLAGS\"  QMAKE_CFLAGS_RELEASE=\"$SLKCFLAGS\" QMAKE_CXXFLAGS_RELEASE=\"$SLKLDFLAGS\"
+# DEFINES+=ENABLE_JIT=0 DEFINES+=ENABLE_YARR_JIT=0 DEFINES+=ENABLE_ASSEMBLER=0
+
+export SLKLDFLAGS="-Wl,-z,relro -Wl,--as-needed"
+
 QTDIR="/usr/lib$LIBDIRSUFFIX/qt" \
 Tools/Scripts/build-webkit \
   --qt \
+  DEFINES+=ENABLE_JIT=0 DEFINES+=ENABLE_YARR_JIT=0 DEFINES+=ENABLE_ASSEMBLER=0 \
   --prefix=/usr/lib${LIBDIRSUFFIX}/qt \
   --release \
   --no-webkit2 \
   --makeargs="$NUMJOBS" \
-  --qmakearg="CONFIG+=production_build QMAKE_CFLAGS=\"$SLKCFLAGS\" QMAKE_CXXFLAGS=\"$SLKCFLAGS\"" \
+  --qmakearg="CONFIG+=production_build QMAKE_LFLAGS_RELEASE=\"$SLKLDFLAGS\" QMAKE_CFLAGS=\"$SLKCFLAGS\" QMAKE_CFLAGS_RELEASE=\"$SLKCFLAGS\" QMAKE_CXXFLAGS_RELEASE=\"$SLKLDFLAGS\" QMAKE_CXXFLAGS=\"$SLKCFLAGS\"" \
   $WEBKITFLAGS || exit 1
 
 # To add support for WebP images add this to --qmakearg
