--- a/fuse.SlackBuild	2021-02-13 03:21:28.343514973 +0200
+++ b/fuse.SlackBuild	2021-02-13 03:21:56.015499229 +0200
@@ -61,6 +61,12 @@
 elif [ "$ARCH" = "x86_64" ]; then
   SLKCFLAGS="-O2 -fPIC"
   LIBDIRSUFFIX="64"
+elif [ "$ARCH" = "aarch64" ]; then
+  SLKCFLAGS="-O2 -fPIC"
+  LIBDIRSUFFIX="64"
+elif [ "$ARCH" = "riscv64" ]; then
+  SLKCFLAGS="-O2 -fPIC"
+  LIBDIRSUFFIX="64"
 else
   SLKCFLAGS="-O2"
   LIBDIRSUFFIX=""
@@ -79,6 +85,10 @@
  \( -perm 666 -o -perm 664 -o -perm 600 -o -perm 444 -o -perm 440 -o -perm 400 \) \
  -exec chmod 644 {} \+
 
+for fp in $CWD/*.patch.xz; do
+  xzcat ${fp} | patch -p1 --verbose || exit 1
+done
+
 CFLAGS="$SLKCFLAGS" \
 CXXFLAGS="$SLKCFLAGS" \
 ./configure \
@@ -94,6 +104,7 @@
   --enable-lib \
   --enable-util \
   --disable-static \
+  --disable-example \
   --build=${ARCH}-slackware-linux || exit 1
 
 make $NUMJOBS || make || exit 1
@@ -151,4 +162,3 @@
 
 cd $PKG
 /sbin/makepkg -l y -c n $TMP/$PKGNAM-$VERSION-$ARCH-$BUILD.txz
-
