--- a/libunwind.SlackBuild	2018-01-23 18:23:13.440286024 +0000
+++ b/libunwind.SlackBuild	2018-01-23 18:23:32.048286024 +0000
@@ -53,6 +53,9 @@
 elif [ "$ARCH" = "armv7hl" ]; then
   SLKCFLAGS="-O3 -march=armv7-a -mfpu=vfpv3-d16"
   LIBDIRSUFFIX=""
+elif [ "$ARCH" = "aarch64" ]; then
+  SLKCFLAGS="-O2"
+  LIBDIRSUFFIX="64"
 else
   SLKCFLAGS="-O2"
   LIBDIRSUFFIX=""
@@ -77,6 +80,10 @@
   \( -perm 666 -o -perm 664 -o -perm 600 -o -perm 444 -o -perm 440 -o -perm 400 \) \
   -exec chmod 644 {} \;
 
+xzcat $CWD/0001-Default-to-enable-debug-frame-also-on-aarch64.patch.xz | patch -p1 --verbose || exit 1
+xzcat $CWD/0001-libunwind-aarch64-fix-build-failure-due-to-asm.patch.xz | patch -p1 --verbose || exit 1
+xzcat $CWD/0001-libunwind-arm-fix-build-failure-due-to-asm.patch.xz | patch -p1 --verbose || exit 1
+
 # Configure:
 CFLAGS="$SLKCFLAGS" \
 ./configure \
