--- a/texlive.SlackBuild	2021-04-20 05:20:00.000000000 +0000
+++ b/texlive.SlackBuild	2021-07-08 19:12:56.039281516 +0000
@@ -60,6 +60,12 @@
 if [ "$ARCH" = "i586" ]; then
   SLKCFLAGS="-O2 -march=i586 -mtune=i686"
   LIBDIRSUFFIX=""
+elif [ "$ARCH" = "aarch64" ]; then
+  SLKCFLAGS="-O2 -fPIC"
+  LIBDIRSUFFIX="64"
+elif [ "$ARCH" = "riscv64" ]; then
+  SLKCFLAGS="-O2 -fPIC"
+  LIBDIRSUFFIX="64"
 elif [ "$ARCH" = "x86_64" ]; then
   SLKCFLAGS="-O2 -fPIC"
   LIBDIRSUFFIX="64"
@@ -68,6 +74,8 @@
   LIBDIRSUFFIX=""
 fi
 
+[[ $ARCH == "riscv64" ]] && OPTIONS="--disable-luajittex --disable-luajithbtex --disable-mfluajit"
+
 rm -rf $PKG
 mkdir -p $TMP $PKG
 cd $TMP
@@ -80,6 +88,9 @@
 # fixes from stable branch
 cat $CWD/patches/fixes-210420.patch | patch -p1 --verbose || exit 1
 
+# broken tests for mendexk on arm
+xzcat $CWD/fix-test-mendex-arm.patch.xz | patch -p1 --verbose || exit 1
+
 mkdir build
 cd build
   CFLAGS="$SLKCFLAGS" \
@@ -121,6 +132,7 @@
     --with-system-ncurses \
     --with-system-harfbuzz \
     --disable-linked-scripts \
+    $OPTIONS \
     --build=$ARCH-slackware-linux || exit 1
 
   make $NUMJOBS || make || exit 1
