--- a/texlive.SlackBuild	2021-04-17 18:41:42.890808788 +0000
+++ b/texlive.SlackBuild	2021-04-22 18:05:31.722315667 +0000
@@ -59,6 +59,12 @@
 if [ "$ARCH" = "i586" ]; then
   SLKCFLAGS="-O2 -march=i586 -mtune=i686"
   LIBDIRSUFFIX=""
+elif [ "$ARCH" = "aarch64" ]; then
+  SLKCFLAGS="-O2 -fPIC"
+  LIBDIRSUFFIX="64"
+elif [ "$ARCH" = "riscv64" ]; then
+  SLKCFLAGS="-O2 -fPIC"
+  LIBDIRSUFFIX="64"
 elif [ "$ARCH" = "x86_64" ]; then
   SLKCFLAGS="-O2 -fPIC"
   LIBDIRSUFFIX="64"
@@ -80,6 +86,8 @@
   POPPLER="--with-system-poppler --with-system-xpdf"
 fi
 
+[[ $ARCH == "riscv64" ]] && OPTIONS="--disable-luajittex --disable-luajithbtex --disable-mfluajit"
+
 rm -rf $PKG
 mkdir -p $TMP $PKG
 cd $TMP
@@ -147,6 +155,7 @@
     --disable-dump-share \
     --disable-linked-scripts \
     $POPPLER \
+    $OPTIONS \
     --build=$ARCH-slackware-linux || exit 1
 
   make $NUMJOBS || make || exit 1
