--- a/cyrus-sasl.SlackBuild	2021-02-13 01:29:38.881735127 +0200
+++ b/cyrus-sasl.SlackBuild	2021-02-13 01:30:18.785712156 +0200
@@ -58,6 +58,12 @@
 elif [ "$ARCH" = "x86_64" ]; then
   SLKCFLAGS="-O2 -fPIC"
   LIBDIRSUFFIX="64"
+elif [ "$ARCH" = "aarch64" ]; then
+  SLKCFLAGS="-O2 -fPIC"
+  LIBDIRSUFFIX="64"
+elif [ "$ARCH" = "riscv64" ]; then
+  SLKCFLAGS="-O2 -fPIC"
+  LIBDIRSUFFIX="64"
 else
   SLKCFLAGS="-O2"
   LIBDIRSUFFIX=""
@@ -93,7 +99,8 @@
   --with-sqlite3=/usr \
   --with-gdbm \
   --with-devrandom=/dev/urandom \
-  --with-dblib=gdbm || exit 1
+  --with-dblib=gdbm \
+  --build=$ARCH-slackware-linux || exit
 
 # How stupid that I need to specify 'sasldir' again for 'make' or else you get
 # the warning "Plugins are being installed into /usr/lib/sasl2, but the library
