--- a/glib.SlackBuild	2017-09-23 11:10:49.595941931 +0100
+++ b/glib.SlackBuild	2017-09-23 11:25:27.368941931 +0100
@@ -50,6 +50,9 @@
 elif [ "$ARCH" = "x86_64" ]; then
   SLKCFLAGS="-O2 -fPIC"
   LIBDIRSUFFIX="64"
+elif [ "$ARCH" = "aarch64" ]; then
+  SLKCFLAGS="-O2"
+  LIBDIRSUFFIX="64"
 else
   SLKCFLAGS="-O2"
   LIBDIRSUFFIX=""
@@ -61,6 +64,7 @@
 cd ${PKGNAM}-$VERSION
 
 bzcat $CWD/glib1.2_1.2.10-17.diff.bz2 | patch -p1 --verbose || exit 1
+xzcat $CWD/glib-1.2.10-gcc5.patch.xz | patch -p0 --verbose || exit 1
 
 # Make sure ownerships and permissions are sane:
 chown -R root:root .
@@ -79,6 +83,7 @@
   --mandir=/usr/man \
   --program-prefix= \
   --program-suffix= \
+  --host=$ARCH-slackware-linux \
   --build=$ARCH-slackware-linux
 
 # Build and install:
