--- protocols/jabber/libiris/src/xmpp/base64/base64.cpp.orig	2017-04-24 10:45:25.102557713 +0100
+++ protocols/jabber/libiris/src/xmpp/base64/base64.cpp	2017-04-24 10:46:27.719557743 +0100
@@ -6,7 +6,7 @@
 {
 	int i;
 	int len = s.size();
-	char tbl[] = "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789+/=";
+	signed char tbl[] = "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789+/=";
 	int a, b, c;
 
 	QByteArray p;
@@ -45,7 +45,7 @@
 	// 64 specifies eof
 	// everything else specifies data
 
-	char tbl[] = {
+	signed char tbl[] = {
 		-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,
 		-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,
 		-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,62,-1,-1,-1,63,
