--- a/scim-tables.SlackBuild	2017-10-01 23:14:25.620747191 +0300
+++ b/scim-tables.SlackBuild	2017-10-01 23:23:41.556747191 +0300
@@ -65,6 +65,9 @@
 elif [ "$ARCH" = "armel" ]; then
   SLKCFLAGS="-O2 -march=armv4t"
   LIBDIRSUFFIX=""
+elif [ "$ARCH" = "aarch64" ]; then
+  SLKCFLAGS="-O2"
+  LIBDIRSUFFIX="64"
 fi
 
 CWD=$(pwd)
@@ -86,6 +89,9 @@
 
 echo Building ...
 
+xzcat $CWD/scim-tables-0.5.14-bz217639.patch.xz | patch -p1 --verbose || exit 1
+xzcat $CWD/scim-tables-0.5.14-bz232860.patch.xz | patch -p1 --verbose || exit 1
+
 ./bootstrap
 
 CFLAGS="$SLKCFLAGS" \
@@ -101,6 +107,9 @@
   --program-suffix= \
   --build=$ARCH-slackware-linux
 
+# Fix source compilation error
+sed -i '698s/#if /#ifdef /g' src/scim_table_imengine_setup.cpp
+
 make $NUMJOBS || make || exit 1
 make DESTDIR=$PKG install || exit 1
 
@@ -129,4 +138,3 @@
 # Build the package:
 cd $PKG
 /sbin/makepkg -l y -c n $TMP/${PKGNAM}-${VERSION}-${ARCH}-${BUILD}.txz
-
