--- a/strigi.SlackBuild	2017-09-27 21:42:06.352619209 +0300
+++ b/strigi.SlackBuild	2017-09-27 21:42:32.050619209 +0300
@@ -54,6 +54,9 @@
 elif [ "$ARCH" = "armv7hl" ]; then
   SLKCFLAGS="-O2 -march=armv7-a -mfpu=vfpv3-d16 -std=gnu++98"
   LIBDIRSUFFIX=""
+elif [ "$ARCH" = "aarch64" ]; then
+  SLKCFLAGS="-O2 -std=gnu++98"
+  LIBDIRSUFFIX="64"
 else
   SLKCFLAGS="-O2 -std=gnu++98"
   LIBDIRSUFFIX=""
