--- a/python.SlackBuild	2017-09-20 21:28:00.180041538 +0300
+++ b/python.SlackBuild	2017-09-20 21:30:39.839041538 +0300
@@ -57,6 +57,8 @@
   LIBDIRSUFFIX=""
 elif [ "$ARCH" = "armel" ]; then
   LIBDIRSUFFIX=""
+elif [ "$ARCH" = "aarch64" ]; then
+  LIBDIRSUFFIX="64"
 fi
 
 # Location for Python site-packages:
@@ -73,11 +75,13 @@
 # We don't want a large libpython*.a:
 zcat $CWD/python.no-static-library.diff.gz | patch -p1 --verbose || exit 1
 
-if [ "$ARCH" = "x86_64" ]; then
-  # Install to lib64 instead of lib and 
-  # Python must report /usr/lib64/python2.7/site-packages as python_lib_dir:
-  zcat $CWD/python.x86_64.diff.gz | patch -p1 --verbose || exit 1
-fi
+case "$ARCH" in
+  x86_64|aarch64)
+    # Install to lib64 instead of lib and 
+    # Python must report /usr/lib64/python2.7/site-packages as python_lib_dir:
+    zcat $CWD/python.x86_64.diff.gz | patch -p1 --verbose || exit 1
+    ;;
+esac
 
 chown -R root:root .
 find . \
