--- a/lensfun.SlackBuild	2022-09-29 21:36:43.793881973 +0300
+++ b/lensfun.SlackBuild	2022-10-01 13:29:40.172870016 +0300
@@ -60,6 +60,12 @@ elif [ "$ARCH" = "s390" ]; then
 elif [ "$ARCH" = "x86_64" ]; then
   SLKCFLAGS="-O2 -fPIC"
   LIBDIRSUFFIX="64"
+elif [ "$ARCH" = "aarch64" ]; then
+  SLKCFLAGS="-O2 -fPIC"
+  LIBDIRSUFFIX="64"
+elif [ "$ARCH" = "riscv64" ]; then
+  SLKCFLAGS="-O2 -fPIC"
+  LIBDIRSUFFIX="64"
 elif [ "$ARCH" = "armv7hl" ]; then
   SLKCFLAGS="-O3 -march=armv7-a -mfpu=vfpv3-d16"
   LIBDIRSUFFIX=""
@@ -93,13 +99,13 @@ cd cmake-build
     -DCMAKE_BUILD_TYPE=Release \
     -DCMAKE_C_FLAGS:STRING="$SLKCFLAGS" \
     -DCMAKE_C_FLAGS_RELEASE:STRING="$SLKCFLAGS" \
-    -DCMAKE_CXX_FLAGS:STRING="$SLKCFLAGS" \
+    -DCMAKE_CXX_FLAGS:STRING="$SLKCFLAGS -fno-delete-null-pointer-checks" \
     -DCMAKE_CXX_FLAGS_RELEASE:STRING="$SLKCFLAGS" \
     -DCMAKE_INSTALL_PREFIX=/usr \
     -DLIB_SUFFIX=${LIBDIRSUFFIX} \
     -DBUILD_LENSTOOL=ON \
-    -DBUILD_FOR_SSE=ON \
-    -DBUILD_FOR_SSE2=ON \
+    -DBUILD_FOR_SSE=OFF \
+    -DBUILD_FOR_SSE2=OFF \
     -DBUILD_TESTS=OFF \
     -DINSTALL_HELPER_SCRIPTS=ON \
     .. || exit 1
