--- a/htdig.SlackBuild	2021-02-13 01:35:44.361516258 +0200
+++ b/htdig.SlackBuild	2021-02-13 01:36:32.689485934 +0200
@@ -59,6 +59,12 @@
 elif [ "$ARCH" = "x86_64" ]; then
   SLKCFLAGS="-O2 -fPIC"
   LIBDIRSUFFIX="64"
+elif [ "$ARCH" = "aarch64" ]; then
+  SLKCFLAGS="-O2 -fPIC"
+  LIBDIRSUFFIX="64"
+elif [ "$ARCH" = "riscv64" ]; then
+  SLKCFLAGS="-O2 -fPIC"
+  LIBDIRSUFFIX="64"
 else
   SLKCFLAGS="-O2"
   LIBDIRSUFFIX=""
@@ -83,6 +89,8 @@
 # Fix "common dir" location and add "synonym_dictionary" entry:
 zcat $CWD/htdig.conf.diff.gz | patch -p1 --verbose || exit 1
 
+xzcat $CWD/htdig-3.2.0b6-buildfix.patch.xz | patch -p1 --verbose || exit 1
+
 # Configure:
 CFLAGS="$SLKCFLAGS" \
 CXXFLAGS="$SLKCFLAGS -std=gnu++98 -fpermissive -fno-delete-null-pointer-checks" \
