--- a/python3.SlackBuild	2020-10-24 23:10:57.674261095 +0000
+++ b/python3.SlackBuild	2020-10-25 20:00:44.789185727 +0000
@@ -68,6 +68,8 @@
   LIBDIRSUFFIX=""
 elif [ "$ARCH" = "armel" ]; then
   LIBDIRSUFFIX=""
+elif [ "$ARCH" = "aarch64" ]; then
+  LIBDIRSUFFIX="64"
 fi
 
 # Location for Python site-packages:
@@ -85,7 +87,7 @@
 # We don't want a large libpython*.a.
 zcat $CWD/python3.no-static-library.diff.gz | patch -p1 --verbose || exit 1
 
-if [ "$ARCH" = "x86_64" ]; then
+if [[ "$ARCH" = "x86_64" || "$ARCH" = "aarch64" ]]; then
   # Install to lib64 instead of lib.
   zcat $CWD/python.sysconfig.py.x86_64.diff.gz | patch -p1 --verbose || exit 1
   zcat $CWD/python3.distutils.x86_64.diff.gz | patch -p1 --verbose || exit 1
