--- a/cyrus-sasl.SlackBuild	2017-11-17 21:44:25.879736598 +0200
+++ b/cyrus-sasl.SlackBuild	2017-11-17 21:45:53.334736598 +0200
@@ -47,6 +47,9 @@
 elif [ "$ARCH" = "x86_64" ]; then
   SLKCFLAGS="-O2 -fPIC"
   LIBDIRSUFFIX="64"
+elif [ "$ARCH" = "aarch64" ]; then
+  SLKCFLAGS="-O2"
+  LIBDIRSUFFIX="64"
 else
   SLKCFLAGS="-O2"
   LIBDIRSUFFIX=""
@@ -81,7 +84,8 @@
   --without-ldap \
   --with-saslauthd \
   --with-gdbm \
-  --with-dblib=gdbm || exit 1
+  --with-dblib=gdbm \
+  --build=$ARCH-slackware-linux || exit 1
 
 # How stupid that I need to specify 'sasldir' again for 'make' or else you get
 # the warning "Plugins are being installed into /usr/lib/sasl2, but the library
