--- a/scim.SlackBuild	2017-10-01 22:48:38.439144397 +0300
+++ b/scim.SlackBuild	2017-10-01 22:49:53.564844882 +0300
@@ -63,6 +63,9 @@
 elif [ "$ARCH" = "armel" ]; then
   SLKCFLAGS="-O2 -march=armv4t"
   LIBDIRSUFFIX=""
+elif [ "$ARCH" = "aarch64" ]; then
+  SLKCFLAGS="-O2"
+  LIBDIRSUFFIX="64"
 fi
 
 NUMJOBS=${NUMJOBS:-" -j7 "}
@@ -79,6 +82,8 @@
 tar xvf $CWD/${PKGNAM}-${VERSION}.tar.?z* || exit 1
 cd ${PKGNAM}-${VERSION} || exit 1
 
+xzcat $CWD/scim-1.4.17-fixes-send-function-call.patch.xz | patch -p1 --verbose || exit 1
+
 ./bootstrap
 
 chown -R root:root .
