--- a/cryptopp.SlackBuild	2021-02-13 03:14:31.779752182 +0200
+++ b/cryptopp.SlackBuild	2021-02-13 03:16:51.987672659 +0200
@@ -60,6 +60,14 @@
   SLKCFLAGS="-O2 -fPIC"
   SLKLDFLAGS="-L/usr/lib64"
   LIBDIRSUFFIX="64"
+elif [ "$ARCH" = "aarch64" ]; then
+  SLKCFLAGS="-O2 -fPIC"
+  SLKLDFLAGS="-L/usr/lib64"
+  LIBDIRSUFFIX="64"
+elif [ "$ARCH" = "riscv64" ]; then
+  SLKCFLAGS="-O2 -fPIC"
+  SLKLDFLAGS="-L/usr/lib64"
+  LIBDIRSUFFIX="64"
 elif [ "$ARCH" = "armv7hl" ]; then
   SLKCFLAGS="-O3 -march=armv7-a -mfpu=vfpv3-d16"
   SLKLDFLAGS=""
