--- a/cryptopp.SlackBuild	2020-10-27 19:12:50.803057531 +0000
+++ b/cryptopp.SlackBuild	2020-11-04 04:48:51.525126206 +0000
@@ -60,6 +60,10 @@
   SLKCFLAGS="-O2 -fPIC"
   SLKLDFLAGS="-L/usr/lib64"
   LIBDIRSUFFIX="64"
+elif [ "$ARCH" = "aarch64" ]; then
+  SLKCFLAGS="-O2 -fPIC"
+  SLKLDFLAGS="-L/usr/lib64"
+  LIBDIRSUFFIX="64"
 elif [ "$ARCH" = "armv7hl" ]; then
   SLKCFLAGS="-O3 -march=armv7-a -mfpu=vfpv3-d16"
   SLKLDFLAGS=""
