--- a/samba.SlackBuild	2021-02-13 02:01:47.108884497 +0200
+++ b/samba.SlackBuild	2021-02-13 02:02:22.388874627 +0200
@@ -52,6 +52,12 @@
   elif [ "$ARCH" = "x86_64" ]; then
     SLKCFLAGS="-O2 -fPIC"
     LIBDIRSUFFIX="64"
+  elif [ "$ARCH" = "aarch64" ]; then
+    SLKCFLAGS="-O2 -fPIC"
+    LIBDIRSUFFIX="64"
+  elif [ "$ARCH" = "riscv64" ]; then
+    SLKCFLAGS="-O2 -fPIC"
+    LIBDIRSUFFIX="64"
   else
     SLKCFLAGS="-O2"
     LIBDIRSUFFIX=""
@@ -109,6 +115,10 @@
   SHADOW_OPTIONS="--without-pam"
 fi
 
+# https://bugzilla.samba.org/show_bug.cgi?id=14100
+sed -e '/HAVE_STROPTS_H/,+2 d' \
+    -i lib/replace/system/network.h
+
 # Some of these options could be auto-detected, but declaring them
 # here doesn't hurt and helps document what features we're trying to
 # build in.
