--- a/samba.SlackBuild	2021-03-13 07:04:05.339362473 +0200
+++ b/samba.SlackBuild	2021-03-13 07:05:14.129102586 +0200
@@ -52,6 +52,12 @@
   elif [ "$ARCH" = "x86_64" ]; then
     SLKCFLAGS="-O2 -fPIC"
     LIBDIRSUFFIX="64"
+  elif [ "$ARCH" = "aarch64" ]; then
+    SLKCFLAGS="-O2 -fPIC"
+    LIBDIRSUFFIX="64"
+  elif [ "$ARCH" = "riscv64" ]; then
+    SLKCFLAGS="-O2 -fPIC"
+    LIBDIRSUFFIX="64"
   else
     SLKCFLAGS="-O2"
     LIBDIRSUFFIX=""
@@ -109,12 +115,16 @@
   SHADOW_OPTIONS="--without-pam"
 fi
 
+# https://bugzilla.samba.org/show_bug.cgi?id=14100
+sed -e '/HAVE_STROPTS_H/,+2 d' \
+    -i lib/replace/system/network.h
+
 # Some of these options could be auto-detected, but declaring them
 # here doesn't hurt and helps document what features we're trying to
 # build in.
 #
 # LDFLAGS are needed to avoid problems with missing symbols.
-LDFLAGS="-Wl,--no-as-needed" \
+LDFLAGS="-Wl,--no-as-needed -lrt -lpthread" \
 CFLAGS="$SLKCFLAGS" \
 ./configure \
   --enable-fhs \
