--- a/qt-nowebkit.SlackBuild	2019-05-05 09:12:00.025050408 +0100
+++ b/qt-nowebkit.SlackBuild	2019-05-05 09:13:57.274658601 +0100
@@ -85,6 +85,9 @@
   # http://bazaar.launchpad.net/~linaro-toolchain-dev/gcc-linaro/4.6/revision/106731
   SLKCFLAGS="-O2 -march=armv7-a -mfpu=vfpv3-d16 -fno-strict-volatile-bitfields"
   LIBDIRSUFFIX=""
+elif [ "$ARCH" = "aarch64" ]; then
+  SLKCFLAGS="-O2 -fno-strict-volatile-bitfields"
+  LIBDIRSUFFIX="64"
 else
   SLKCFLAGS="-O2"
   LIBDIRSUFFIX=""
@@ -165,14 +168,18 @@
 #zcat $CWD/patches/qt-everywhere-opensource-src-4.8.0-tp-multilib-optflags.patch.gz | patch -p1 --verbose || exit 1
 zcat $CWD/patches/qt-everywhere-opensource-src-4.8.7-icu59.patch.gz | patch -p1 --verbose || exit 1
 
+echo "QMAKE_CXXFLAGS += -std=gnu++98" >> src/3rdparty/javascriptcore/JavaScriptCore/JavaScriptCore.pri
+echo "QMAKE_CXXFLAGS += -std=gnu++98" >> src/3rdparty/webkit/Source/JavaScriptCore/JavaScriptCore.pri
+echo "QMAKE_CXXFLAGS += -std=gnu++98" >> src/plugins/accessible/qaccessiblebase.pri
+
 # All other distros I checked build with -no-webkit. But this has the downside of
 # breaking the build when QtWebKit is present on the build system and some things
 # (assistant for one) are built without support for QtWebKit. So let's go with
 # -webkit and remove the built files later from the package. -HW
 
 export CFLAGS="$SLKCFLAGS"
-export CXXFLAGS="$SLKCFLAGS -std=gnu++98 -fpermissive -fno-delete-null-pointer-checks"
-export OPENSOURCE_CXXFLAGS="$SLKCFLAGS -std=gnu++98 -fpermissive -fno-delete-null-pointer-checks"
+export CXXFLAGS="$SLKCFLAGS -fpermissive -fno-delete-null-pointer-checks"
+export OPENSOURCE_CXXFLAGS="$SLKCFLAGS -fpermissive -fno-delete-null-pointer-checks"
 ./configure \
   -confirm-license \
   -opensource \
