--- a/qt-nowebkit.SlackBuild	2018-04-13 05:50:30.000000000 +0300
+++ b/qt-nowebkit.SlackBuild	2018-04-25 21:19:12.501912084 +0300
@@ -85,6 +85,9 @@
   # http://bazaar.launchpad.net/~linaro-toolchain-dev/gcc-linaro/4.6/revision/106731
   SLKCFLAGS="-O2 -march=armv7-a -mfpu=vfpv3-d16 -fno-strict-volatile-bitfields"
   LIBDIRSUFFIX=""
+elif [ "$ARCH" = "aarch64" ]; then
+  SLKCFLAGS="-O2 -fno-strict-volatile-bitfields"
+  LIBDIRSUFFIX="64"
 else
   SLKCFLAGS="-O2"
   LIBDIRSUFFIX=""
@@ -162,14 +165,20 @@
 # that uses them with that C++ standard:
 zcat $CWD/qt.icu59.c++11.diff.gz | patch -p1 --verbose || exit 1
 
+xzcat $CWD/qt-aarch64.patch.xz | patch -p1 --verbose || exit 1
+
+echo "QMAKE_CXXFLAGS += -std=gnu++98" >> src/3rdparty/javascriptcore/JavaScriptCore/JavaScriptCore.pri
+echo "QMAKE_CXXFLAGS += -std=gnu++98" >> src/3rdparty/webkit/Source/JavaScriptCore/JavaScriptCore.pri
+echo "QMAKE_CXXFLAGS += -std=gnu++98" >> src/plugins/accessible/qaccessiblebase.pri
+
 # All other distros I checked build with -no-webkit. But this has the downside of
 # breaking the build when QtWebKit is present on the build system and some things
 # (assistant for one) are built without support for QtWebKit. So let's go with
 # -webkit and remove the built files later from the package. -HW
 
 export CFLAGS="$SLKCFLAGS"
-export CXXFLAGS="$SLKCFLAGS -std=gnu++98 -fpermissive -fno-delete-null-pointer-checks"
-export OPENSOURCE_CXXFLAGS="$SLKCFLAGS -std=gnu++98 -fpermissive -fno-delete-null-pointer-checks"
+export CXXFLAGS="$SLKCFLAGS -fpermissive -fno-delete-null-pointer-checks"
+export OPENSOURCE_CXXFLAGS="$SLKCFLAGS -fpermissive -fno-delete-null-pointer-checks"
 ./configure \
   -confirm-license \
   -opensource \
