--- a/qtwebkit/qtwebkit.SlackBuild	2017-09-23 19:59:05.611808950 +0300
+++ b/qtwebkit/qtwebkit.SlackBuild	2017-09-23 20:02:32.280808950 +0300
@@ -27,7 +27,7 @@
 VERSION=$(ls qtwebkit-*.tar.?z* | rev | cut -f 3- -d . | cut -f 1 -d - | rev)
 BUILD=${BUILD:-1}
 
-NUMJOBS=${NUMJOBS:--j7}
+NUMJOBS=${NUMJOBS:--j5}
 
 # Automatically determine the architecture we're building on:
 MARCH=$( uname -m )
@@ -64,6 +64,10 @@
   SLKCFLAGS="-O2 -march=armv7-a -mfpu=vfpv3-d16 -fno-strict-volatile-bitfields"
   LIBDIRSUFFIX=""
   WEBKITFLAGS="--no-force-sse2"
+elif [ "$ARCH" = "aarch64" ]; then
+  SLKCFLAGS="-O2"
+  LIBDIRSUFFIX="64"
+  WEBKITFLAGS="--no-force-sse2"
 else
   SLKCFLAGS="-O2"
   LIBDIRSUFFIX=""
@@ -96,14 +100,20 @@
  \( -perm 666 -o -perm 664 -o -perm 600 -o -perm 444 -o -perm 440 -o -perm 400 \) \
  -exec chmod 644 {} \;
 
+# Configure & build:
+# For ARM we added:
+# QMAKE_LFLAGS_RELEASE=\"$SLKLDFLAGS\"  QMAKE_CFLAGS_RELEASE=\"$SLKCFLAGS\" QMAKE_CXXFLAGS_RELEASE=\"$SLKLDFLAGS\"
+# DEFINES+=ENABLE_JIT=0 DEFINES+=ENABLE_YARR_JIT=0 DEFINES+=ENABLE_ASSEMBLER=0
+
 QTDIR="/usr/lib$LIBDIRSUFFIX/qt" \
 Tools/Scripts/build-webkit \
   --qt \
+  DEFINES+=ENABLE_JIT=0 DEFINES+=ENABLE_YARR_JIT=0 DEFINES+=ENABLE_ASSEMBLER=0 \
   --prefix=/usr/lib${LIBDIRSUFFIX}/qt \
   --release \
   --no-webkit2 \
   --makeargs="$NUMJOBS" \
-  --qmakearg="CONFIG+=production_build QMAKE_CFLAGS=\"$SLKCFLAGS\" QMAKE_CXXFLAGS=\"$SLKCFLAGS\"" \
+  --qmakearg="CONFIG+=production_build QMAKE_LFLAGS_RELEASE=\"$SLKLDFLAGS\" QMAKE_CFLAGS=\"$SLKCFLAGS\" QMAKE_CFLAGS_RELEASE=\"$SLKCFLAGS\" QMAKE_CXXFLAGS_RELEASE=\"$SLKLDFLAGS\" QMAKE_CXXFLAGS=\"$SLKCFLAGS\"" \
   $WEBKITFLAGS || exit 1
 
 # To add support for WebP images add this to --qmakearg
