--- a/chmlib.SlackBuild	2021-02-13 03:09:39.431901929 +0200
+++ b/chmlib.SlackBuild	2021-02-13 03:11:56.615831693 +0200
@@ -59,6 +59,12 @@
 elif [ "$ARCH" = "x86_64" ]; then
   SLKCFLAGS="-O2 -fPIC"
   LIBDIRSUFFIX="64"
+elif [ "$ARCH" = "aarch64" ]; then
+  SLKCFLAGS="-O2 -fPIC"
+  LIBDIRSUFFIX="64"
+elif [ "$ARCH" = "riscv64" ]; then
+  SLKCFLAGS="-O2 -fPIC"
+  LIBDIRSUFFIX="64"
 else
   SLKCFLAGS="-O2"
   LIBDIRSUFFIX=""
@@ -77,6 +83,9 @@
  \( -perm 666 -o -perm 664 -o -perm 600 -o -perm 444 -o -perm 440 -o -perm 400 \) \
  -exec chmod 644 {} \+
 
+xzcat $CWD/chmlib-0001-Patch-to-fix-integer-types-problem-by-Goswin-von-Bre.patch.xz || patch -p1 --verbose || exit 1
+xzcat $CWD/chmlib-0002-Fix-for-extract_chmLib-confusing-empty-files-with-di.patch.xz || patch -p1 --verbose || exit 1
+
 CFLAGS="$SLKCFLAGS" \
 CXXFLAGS="$SLKCFLAGS" \
 ./configure \
