--- a/icu4c.SlackBuild	2021-02-13 03:29:00.307258808 +0200
+++ b/icu4c.SlackBuild	2021-02-13 03:29:51.359229999 +0200
@@ -62,6 +62,12 @@
 elif [ "$ARCH" = "x86_64" ]; then
   SLKCFLAGS="-O2 -fPIC"
   LIBDIRSUFFIX="64"
+elif [ "$ARCH" = "aarch64" ]; then
+  SLKCFLAGS="-O2 -fPIC"
+  LIBDIRSUFFIX="64"
+elif [ "$ARCH" = "riscv64" ]; then
+  SLKCFLAGS="-O2 -fPIC"
+  LIBDIRSUFFIX="64"
 else
   SLKCFLAGS="-O2"
   LIBDIRSUFFIX=""
@@ -81,8 +87,10 @@
   -o -perm 400 \) -exec chmod 644 {} \+
 
 cd source/
+  CC=gcc \
+  CXX=g++ \
   CFLAGS="$SLKCFLAGS" \
-  CXXFLAGS="$SLKCFLAGS" \
+  CXXFLAGS="$SLKCFLAGS -std=c++11" \
   ./configure \
     --prefix=/usr \
     --libdir=/usr/lib${LIBDIRSUFFIX} \
