--- a/python2.SlackBuild	2021-02-10 21:02:27.008082346 +0200
+++ b/python2.SlackBuild	2021-02-10 21:04:24.943985531 +0200
@@ -65,6 +65,10 @@
   LIBDIRSUFFIX=""
 elif [ "$ARCH" = "armel" ]; then
   LIBDIRSUFFIX=""
+elif [ "$ARCH" = "aarch64" ]; then
+  LIBDIRSUFFIX="64"
+elif [ "$ARCH" = "riscv64" ]; then
+  LIBDIRSUFFIX="64"
 fi
 
 # Location for Python site-packages:
@@ -81,7 +85,7 @@
 # We don't want a large libpython*.a:
 zcat $CWD/python.no-static-library.diff.gz | patch -p1 --verbose || exit 1
 
-if [ "$ARCH" = "x86_64" ]; then
+if [ "$ARCH" = "x86_64" -o "$ARCH" = "aarch64" -o "$ARCH" = "riscv64" ]; then
   # Install to lib64 instead of lib and 
   # Python must report /usr/lib64/python2.7/site-packages as python_lib_dir:
   zcat $CWD/python.x86_64.diff.gz | patch -p1 --verbose || exit 1
