--- a/ispell.SlackBuild	2017-09-20 22:25:36.683174937 +0300
+++ b/ispell.SlackBuild	2017-09-20 22:26:08.900174937 +0300
@@ -38,11 +38,10 @@
 TMP=${TMP:-/tmp}
 PKG=$TMP/package-ispell
 
-if [ "$ARCH" = "x86_64" ]; then
-  LIBDIRSUFFIX="64"
-else
-  LIBDIRSUFFIX=""
-fi
+case "$ARCH" in
+  x86_64|aarch64) LIBDIRSUFFIX="64" ;;
+               *) LIBDIRSUFFIX="" ;;
+esac
 
 rm -rf $PKG
 mkdir -p $PKG $PKG
