--- a/etc.SlackBuild	2021-02-10 17:17:19.841971438 +0200
+++ b/etc.SlackBuild	2021-02-10 17:18:04.997939374 +0200
@@ -57,6 +57,10 @@
 # Fix paths if lib64 is needed:
 if [ "$ARCH" = "x86_64" ]; then
   sed -i -e "s#/lib#/lib64#" etc/ld.so.conf.new
+elif [ "$ARCH" = "aarch64" ]; then
+  sed -i -e "s#/lib#/lib64#" etc/ld.so.conf.new
+elif [ "$ARCH" = "riscv64" ]; then
+  sed -i -e "s#/lib#/lib64#" etc/ld.so.conf.new
 fi
 zcat $CWD/nsswitch.conf.gz > $PKG/etc/nsswitch.conf.new
 
