--- a/glew.SlackBuild	2021-02-13 00:04:50.661703925 +0200
+++ b/glew.SlackBuild	2021-02-13 00:05:25.049698537 +0200
@@ -60,6 +60,12 @@
 elif [ "$ARCH" = "x86_64" ]; then
   SLKCFLAGS="-O2 -fPIC"
   LIBDIRSUFFIX="64"
+elif [ "$ARCH" = "aarch64" ]; then
+  SLKCFLAGS="-O2 -fPIC"
+  LIBDIRSUFFIX="64"
+elif [ "$ARCH" = "riscv64" ]; then
+  SLKCFLAGS="-O2 -fPIC"
+  LIBDIRSUFFIX="64"
 else
   SLKCFLAGS="-O2"
   LIBDIRSUFFIX=""
@@ -89,6 +95,8 @@
 # Use lib${LIBDIRSUFFIX} (if needed) in Makefile:
 sed -i "s,/lib,/lib${LIBDIRSUFFIX},g" Makefile
 
+sed -i "s,\/lib\(\s\|$\),\/lib${LIBDIRSUFFIX}\1,g" config/Makefile.linux
+
 make $NUMJOBS OPT="$SLKCFLAGS" || make OPT="$SLKCFLAGS" || exit 1
 make install.all GLEW_DEST=$PKG/usr || exit 1
 
