--- a/t1lib.SlackBuild	2017-09-28 23:49:02.280940209 +0300
+++ b/t1lib.SlackBuild	2017-09-28 23:50:01.949952798 +0300
@@ -44,6 +44,9 @@
 elif [ "$ARCH" = "x86_64" ]; then
   SLKCFLAGS="-O2 -fPIC"
   LIBDIRSUFFIX="64"
+elif [ "$ARCH" = "aarch64" ]; then
+  SLKCFLAGS="-O2"
+  LIBDIRSUFFIX="64"
 fi
 
 NUMJOBS=${NUMJOBS:-" -j7 "}
@@ -87,7 +90,7 @@
   --enable-static=no \
   --build=$ARCH-slackware-linux
 
-make $NUMJOBS || make || exit 1
+make $NUMJOBS without_doc || make without_doc || exit 1
 make install prefix=$PKG/usr libdir=$PKG/usr/lib${LIBDIRSUFFIX} || exit 1
 
 find $PKG | xargs file | grep -e "executable" -e "shared object" | grep ELF \
