--- a/t1lib.SlackBuild	2021-02-13 03:53:38.606409605 +0200
+++ b/t1lib.SlackBuild	2021-02-13 03:54:09.298389092 +0200
@@ -53,6 +53,12 @@
 elif [ "$ARCH" = "x86_64" ]; then
   SLKCFLAGS="-O2 -fPIC"
   LIBDIRSUFFIX="64"
+elif [ "$ARCH" = "aarch64" ]; then
+  SLKCFLAGS="-O2 -fPIC"
+  LIBDIRSUFFIX="64"
+elif [ "$ARCH" = "riscv64" ]; then
+  SLKCFLAGS="-O2 -fPIC"
+  LIBDIRSUFFIX="64"
 fi
 
 NUMJOBS=${NUMJOBS:-" -j$(expr $(nproc) + 1) "}
@@ -95,7 +101,7 @@
   --enable-static=no \
   --build=$ARCH-slackware-linux || exit 1
 
-make $NUMJOBS || make || exit 1
+make $NUMJOBS without_doc || make without_doc || exit 1
 make install prefix=$PKG/usr libdir=$PKG/usr/lib${LIBDIRSUFFIX} || exit 1
 
 # Don't ship .la files:
