--- a/seamonkey.SlackBuild	2018-04-04 22:30:26.772463980 +0300
+++ b/seamonkey.SlackBuild	2018-04-04 22:35:45.185463980 +0300
@@ -56,6 +56,9 @@
 elif [ "$ARCH" = "x86_64" ]; then
   LIBDIRSUFFIX="64"
   OPTIMIZE_FLAG=${OPTIMIZE_FLAG:-"-O1"}
+elif [ "$ARCH" = "aarch64" ]; then
+  LIBDIRSUFFIX="64"
+  OPTIMIZE_FLAG=${OPTIMIZE_FLAG:-"-O1"}
 else
   LIBDIRSUFFIX=""
   OPTIMIZE_FLAG=${OPTIMIZE_FLAG:-"-O1"}
@@ -102,6 +105,21 @@
 rm -rf seamonkey-unpack
 cd seamonkey-${TARBALLVER} || exit 1
 
+# Apply patches for ARM:
+# All from Fedora:
+xzcat $CWD/xulrunner-27.0-build-arm.patch.xz | sed 's?/mozilla-release/?/mozilla/?g' | patch -p1 --verbose || exit 1
+
+pushd mozilla
+xzcat $CWD/firefox-48-mozilla-256180.patch.xz | patch -p1 --verbose || exit 1
+xzcat $CWD/firefox-52-mozilla-revert-1148544.patch.xz | patch -p1 --verbose || exit 1
+xzcat $CWD/firefox-50-mozilla-440908.patch.xz | patch -p1 --verbose || exit 1
+xzcat $CWD/firefox-52-build-prbool.patch.xz | patch -p1 --verbose || exit 1
+xzcat $CWD/firefox-51-mozilla-1005640.patch.xz | patch -p1 --verbose || exit 1
+xzcat $CWD/firefox-51-mozilla-1321579.patch.xz | patch -p1 --verbose || exit 1
+xzcat $CWD/seamonkey-2.46-exthandler.patch.xz | sed 's?/comm-release/?/?g' | patch -p2 --verbose || exit 1
+xzcat $CWD/firefox-52-rhbz-1451055.patch.xz | patch -p1 --verbose || exit 1
+popd
+
 # Retain GTK+ v2 scrolling behavior:
 zcat $CWD/sm.ui.scrollToClick.diff.gz | patch -p1 --verbose || exit 1
 
@@ -119,6 +137,15 @@
 OPTIONS="\
   --prefix=/usr \
   --libdir=/usr/lib${LIBDIRSUFFIX} \
+  \
+  --enable-system-sqlite \
+  --with-system-libvpx \
+  --enable-system-ffi \
+  --without-system-icu \
+  --enable-system-cairo \
+  --disable-stylo \
+  --disable-webrtc \
+  \
   --enable-cpp-rtti \
   --enable-default-toolkit=cairo-gtk3 \
   --enable-startup-notification \
@@ -140,11 +167,20 @@
   --enable-application=suite \
   --with-pthreads \
   --host=$ARCH-slackware-linux \
-  --target=$ARCH-slackware-linux"
+  --target=$ARCH-slackware-linux \
+  "
+
+export SLKCFLAGS="$SLKCFLAGS -g -Wformat -fexceptions -fPIC -Wl,-z,relro -Wl,-z,now"
+export SLKLDFLAGS="-Wl,--no-as-needed -Wl,--no-keep-memory"
 
 export BUILD_OFFICIAL=1
 export MOZILLA_OFFICIAL=1
 export MOZ_MAKE_FLAGS="$NUMJOBS"
+export MOZ_LINK_FLAGS="$SLKLDFLAGS"
+export LDFLAGS="$SLKLDFLAGS"
+export MOZ_OPT_FLAGS="$SLKCFLAGS"
+export CFLAGS="$SLKCFLAGS"
+export CXXFLAGS="$SLKCFLAGS -fno-delete-null-pointer-checks"
 
 # Clear some variables that could break the build
 unset DBUS_SESSION_BUS_ADDRESS ORBIT_SOCKETDIR SESSION_MANAGER \
