--- a/sc.SlackBuild	2017-09-20 22:07:29.631174937 +0300
+++ b/sc.SlackBuild	2017-09-20 22:08:43.529174937 +0300
@@ -34,13 +34,12 @@
   esac
 fi
 
-NUMJOBS=${NUMJOBS:-" -j7 "}
+NUMJOBS=${NUMJOBS:-" -j5 "}
 
-if [ "$ARCH" = "x86_64" ]; then
-  LIBDIRSUFFIX="64"
-else
-  LIBDIRSUFFIX=""
-fi
+case "$ARCH" in
+  x86_64|aarch64) LIBDIRSUFFIX="64" ;;
+               *) LIBDIRSUFFIX="" ;;
+esac
 
 CWD=$(pwd)
 TMP=${TMP:-/tmp}
