--- a/mozilla-firefox.SlackBuild	2019-07-09 20:32:46.500240877 +0100
+++ b/mozilla-firefox.SlackBuild	2019-07-12 21:39:07.581630363 +0100
@@ -132,6 +132,9 @@
 elif [ "$ARCH" = "armel" ]; then
   SLKCFLAGS="-O2 -march=armv4t"
   LIBDIRSUFFIX=""
+elif [ "$ARCH" = "aarch64" ]; then
+  SLKCFLAGS="-O2"
+  LIBDIRSUFFIX="64"
 else
   SLKCFLAGS="-O2"
   LIBDIRSUFFIX=""
@@ -208,8 +211,13 @@
   \( -perm 666 -o -perm 664 -o -perm 600 -o -perm 444 -o -perm 440 -o -perm 400 \) \
   -exec chmod 644 {} \;
 
+xzcat $CWD/Bug-1238661---fix-mozillaSignalTrampoline-to-work-.patch.xz | patch -p1 --verbose || exit 1
+
 # Our building options, in a configure-like display ;)
 OPTIONS="\
+  --host=$ARCH-slackware-linux \
+  --target=$ARCH-slackware-linux \
+  \
   --enable-official-branding \
   --prefix=/usr \
   --libdir=/usr/lib${LIBDIRSUFFIX} \
@@ -284,6 +292,12 @@
 # Add the $OPTIONS above to .mozconfig:
 for option in $OPTIONS; do echo "ac_add_options $option" >> .mozconfig; done
 
+# ARM: see https://bugzilla.redhat.com/show_bug.cgi?id=1426850
+# This isn't added using $SLKCONFARGS because it doesn't cater for things with spaces in:
+# unless I edit the for loop above to use IFS with some other variable.
+# this is easier right now ;-)
+echo 'ac_add_options --enable-optimize="-O1 -fno-schedule-insns"' >> .mozconfig
+
 # https://developer.mozilla.org/en-US/docs/Building_with_Profile-Guided_Optimization
 # Thanks to ArchLinux and Gentoo for the additional hints.
 if [ "$PGO" = "yes" ]; then
