--- a/at.SlackBuild	2017-09-17 14:51:58.439233917 +0100
+++ b/at.SlackBuild	2017-09-17 14:53:46.426233969 +0100
@@ -35,15 +35,23 @@
   esac
 fi
 
-if [ "$ARCH" = "i586" ]; then
-  SLKCFLAGS="-O2 -march=i586 -mtune=i686"
-elif [ "$ARCH" = "s390" ]; then
-  SLKCFLAGS="-O2"
-elif [ "$ARCH" = "x86_64" ]; then
-  SLKCFLAGS="-O2 -fPIC"
-else
-  SLKCFLAGS="-O2"
-fi
+case "$ARCH" in
+  i?86)    SLKCFLAGS="-O2 -march=i486 -mtune=i686"
+           LIBDIRSUFFIX=""
+           ;;
+  s390)    SLKCFLAGS="-O2"
+           LIBDIRSUFFIX=""
+           ;;
+  x86_64)  SLKCFLAGS="-O2 -fPIC"
+           LIBDIRSUFFIX="64"
+           ;;
+  aarch64) SLKCFLAGS="-O2"
+           LIBDIRSUFFIX="64"
+           ;;
+        *) SLKCFLAGS="-O2"
+           LIBDIRSUFFIX=""
+           ;;
+esac
 
 CWD=$(pwd)
 TMP=${TMP:-/tmp}
