--- a/js185.SlackBuild	2012-07-19 01:16:35.000000000 +0300
+++ b/js185.SlackBuild	2017-09-28 21:54:31.503677628 +0300
@@ -54,6 +54,9 @@
 elif [ "$ARCH" = "x86_64" ]; then
   SLKCFLAGS="-O2 -fPIC"
   LIBDIRSUFFIX="64"
+elif [ "$ARCH" = "aarch64" ]; then
+  SLKCFLAGS="-O2"
+  LIBDIRSUFFIX="64"
 else
   SLKCFLAGS="-O2"
   LIBDIRSUFFIX=""
@@ -74,7 +77,21 @@
 
 cd js/src
 
-patch -p0 < $CWD/js185-destdir.patch || exit 1
+xzcat $CWD/js-1.8.5-64bit-big-endian.patch.xz | patch -p4 --verbose || exit 1
+xzcat $CWD/js-1.8.5-secondary-jit.patch.xz | patch -p4 --verbose || exit 1
+xzcat $CWD/js185-destdir.patch.xz | patch -p2 --verbose || exit 1
+xzcat $CWD/js-1.8.5-537701.patch.xz | patch -p3 --verbose || exit 1
+xzcat $CWD/js185-arm-nosoftfp.patch.xz | patch -p3 --verbose || exit 1
+#xzcat $CWD/js185-libedit.patch.xz | patch -p3 --verbose || exit 1
+xzcat $CWD/0001-Make-js-config.h-multiarch-compatible.patch.xz | patch -p3 --verbose || exit 1
+xzcat $CWD/aarch64.patch.xz | patch -p3 --verbose || exit 1
+# bz#1096135
+xzcat $CWD/ppc64le.patch.xz | patch -p3 --verbose || exit 1
+# https://bugzilla.redhat.com/show_bug.cgi?id=1178141 (http://hg.mozilla.org/mozilla-central/rev/a7b220e7425a)
+#xzcat $CWD/js-1.8.5-array-recursion.patch.xz | patch -p3 --verbose || exit 1
+xzcat $CWD/js-1.8.5-c++11.patch.xz | patch -p3 --verbose || exit 1
+# update for 48-bit VA
+xzcat $CWD/mozjs1.8.5-tag.patch.xz | patch -p3 --verbose || exit 1
 
 CXXFLAGS="$SLKCFLAGS" \
 CFLAGS="$SLKCFLAGS" \
@@ -84,8 +101,9 @@
   --sysconfdir=/etc \
   --localstatedir=/var \
   --mandir=/usr/man \
+  --host=$ARCH-slackware-linux \
   --build=$ARCH-slackware-linux \
-  --enable-optimize="-O2" \
+  --disable-optimize \
   --enable-threadsafe \
   --with-system-nspr \
   --with-nspr-prefix=/usr || exit 1
