--- a/rust.SlackBuild	2019-05-26 10:07:57.284347731 +0100
+++ b/rust.SlackBuild	2019-05-26 10:08:53.097147137 +0100
@@ -131,6 +131,9 @@
 elif [ "$ARCH" = "armv7hl" ]; then
   SLKCFLAGS=""
   LIBDIRSUFFIX=""
+elif [ "$ARCH" = "aarch64" ]; then
+  SLKCFLAGS="-O2"
+  LIBDIRSUFFIX="64"
 else
   SLKCFLAGS="-O2"
   LIBDIRSUFFIX=""
@@ -203,6 +206,9 @@
 
 [target.armv7-unknown-linux-gnueabihf]
 llvm-config = "/usr/bin/llvm-config"
+
+[target.aarch64-unknown-linux-gnu]
+llvm-config = "/usr/bin/llvm-config"
 EOF
 fi
 
@@ -241,8 +247,8 @@
 else
   # README.md says gcc 4.7 / clang 3.x or later needed
   # but building fails for me with GCC 5.3 from slackware 14.2
-  export CC=clang
-  export CXX=clang++
+#  export CC=clang
+#  export CXX=clang++
   CFLAGS="$SLKCFLAGS" \
   CXXFLAGS="$SLKCFLAGS" \
   python x.py dist  || exit 1
