--- a/rust.SlackBuild	2018-07-11 10:19:08.504001248 +0300
+++ b/rust.SlackBuild	2018-07-14 09:33:59.294340785 +0300
@@ -126,6 +126,9 @@
 elif [ "$ARCH" = "armv7hl" ]; then
   SLKCFLAGS=""
   LIBDIRSUFFIX=""
+elif [ "$ARCH" = "aarch64" ]; then
+  SLKCFLAGS="-O2"
+  LIBDIRSUFFIX="64"
 else
   SLKCFLAGS="-O2"
   LIBDIRSUFFIX=""
@@ -198,6 +201,9 @@
 
 [target.armv7-unknown-linux-gnueabihf]
 llvm-config = "/usr/bin/llvm-config"
+
+[target.aarch64-unknown-linux-gnu]
+llvm-config = "/usr/bin/llvm-config"
 EOF
 fi
 
@@ -236,8 +242,8 @@
 else
   # README.md says gcc 4.7 / clang 3.x or later needed
   # but building fails for me with GCC 5.3 from slackware 14.2
-  CC=clang \
-  CXX=clang++ \
+#  CC=clang \
+#  CXX=clang++ \
   CFLAGS="$SLKCFLAGS" \
   CXXFLAGS="$SLKCFLAGS" \
   python x.py dist  || exit 1
