--- a/net-snmp.SlackBuild	2018-11-27 09:38:06.099813026 +0000
+++ b/net-snmp.SlackBuild	2018-11-27 09:39:39.342030946 +0000
@@ -58,6 +58,9 @@
 elif [ "$ARCH" = "x86_64" ]; then
   SLKCFLAGS="-O2 -fPIC"
   LIBDIRSUFFIX="64"
+elif [ "$ARCH" = "aarch64" ]; then
+  SLKCFLAGS="-O2"
+  LIBDIRSUFFIX="64"
 else
   SLKCFLAGS="-O2"
   LIBDIRSUFFIX=""
@@ -97,6 +100,10 @@
 #sed -i "s|\$(PYMAKE) install|\$(PYMAKE) install --root=$PKG|" \
 #  Makefile.in
 
+# fix global environment
+unset CPPFLAGS
+unset LDFLAGS
+
 # Configure:
 CFLAGS="$SLKCFLAGS" \
 ./configure \
