--- a/sharutils.SlackBuild	2021-01-20 20:13:08.261005844 +0200
+++ b/sharutils.SlackBuild	2021-02-10 17:25:57.709601820 +0200
@@ -56,6 +56,12 @@
 elif [ "$ARCH" = "x86_64" ]; then
   SLKCFLAGS="-O2 -fcommon"
   LIBDIRSUFFIX="64"
+elif [ "$ARCH" = "aarch64" ]; then
+  SLKCFLAGS="-O2 -fcommon"
+  LIBDIRSUFFIX="64"
+elif [ "$ARCH" = "riscv64" ]; then
+  SLKCFLAGS="-O2 -fcommon"
+  LIBDIRSUFFIX="64"
 fi
 
 # Don't use icecream:
