--- a/metamail.SlackBuild	2021-02-13 01:49:35.365049552 +0200
+++ b/metamail.SlackBuild	2021-02-13 01:50:11.013043373 +0200
@@ -62,6 +62,12 @@
 elif [ "$ARCH" = "armel" ]; then
   SLKCFLAGS="-O2 -march=armv4t"
   LIBDIRSUFFIX=""
+elif [ "$ARCH" = "aarch64" ]; then
+  SLKCFLAGS="-O2 -fPIC"
+  LIBDIRSUFFIX="64"
+elif [ "$ARCH" = "riscv64" ]; then
+  SLKCFLAGS="-O2 -fPIC"
+  LIBDIRSUFFIX="64"
 else
   SLKCFLAGS="-O2"
   LIBDIRSUFFIX=""
@@ -97,6 +103,7 @@
   --disable-static \
   --program-prefix="" \
   --program-suffix="" \
+  --host=$ARCH-slackware-linux \
   --build=$ARCH-slackware-linux || exit 1
 
 ( cd bin
