--- a/mesa.SlackBuild	2018-05-01 12:39:54.909073444 +0300
+++ b/mesa.SlackBuild	2018-05-01 12:42:31.724905567 +0300
@@ -30,8 +30,8 @@
 NUMJOBS=${NUMJOBS:--j7}
 
 # Be sure this list is up-to-date:
-DRI_DRIVERS="i915,i965,nouveau,r200,radeon,swrast"
-GALLIUM_DRIVERS="nouveau,r300,r600,svga,radeonsi,swrast,virgl"
+DRI_DRIVERS="nouveau,swrast"
+GALLIUM_DRIVERS="nouveau,swrast,virgl,vc4"
 EGL_PLATFORMS="drm,x11"
 
 if [ -z "$ARCH" ]; then
@@ -63,6 +63,10 @@
   SLKCFLAGS="-O2 -fPIC"
   LIBDIRSUFFIX="64"
   VULKAN=" --with-vulkan-drivers=intel,radeon "
+elif [ "$ARCH" = "aarch64" ]; then
+  SLKCFLAGS="-O2"
+  LIBDIRSUFFIX="64"
+#  VULKAN=" --with-vulkan-drivers=intel,radeon "
 else
   SLKCFLAGS="-O2"
   LIBDIRSUFFIX=""
@@ -107,6 +111,7 @@
 autoreconf -fi
 
 CFLAGS="$SLKCFLAGS" \
+CXXFLAGS="$SLKCFLAGS -D_GLIBCXX_USE_CXX11_ABI=0" \
 ./configure \
   --prefix=/usr \
   --sysconfdir=/etc \
@@ -134,8 +139,6 @@
   --enable-gles1 \
   --enable-gles2 \
   --enable-vdpau \
-  --enable-opencl \
-  --enable-opencl-icd \
   --build=$ARCH-slackware-linux || exit 1
 
 # This is autodetected anyway:
