--- a/mozjs78.SlackBuild	2021-02-24 18:37:43.246169500 +0200
+++ b/mozjs78.SlackBuild	2021-02-24 18:39:59.374098318 +0200
@@ -62,6 +62,12 @@
 elif [ "$ARCH" = "x86_64" ]; then
   SLKCFLAGS="-O2 -fPIC"
   LIBDIRSUFFIX="64"
+elif [ "$ARCH" = "aarch64" ]; then
+  SLKCFLAGS="-O2 -fPIC"
+  LIBDIRSUFFIX="64"
+elif [ "$ARCH" = "riscv64" ]; then
+  SLKCFLAGS="-O2 -fPIC"
+  LIBDIRSUFFIX="64"
 else
   SLKCFLAGS="-O2"
   LIBDIRSUFFIX=""
@@ -101,6 +107,7 @@
 zcat $CWD/patches/init_patch.patch.gz | patch -p1 --verbose || exit 1
 zcat $CWD/patches/spidermonkey_checks_disable.patch.gz | patch -p1 --verbose || exit 1
 #zcat $CWD/patches/encies.armv7_disable_WASM_EMULATE_ARM_UNALIGNED_FP_ACCESS.patch.gz | patch -p1 --verbose || exit 1
+xzcat $CWD/Add-riscv64-support.patch.xz | patch -p1 --verbose || exit 1
 
 # Remove bundled zlib directory and use system version:
 rm -rf modules/zlib
