--- a/grub.SlackBuild	2023-03-28 18:01:10.378009359 +0000
+++ b/grub.SlackBuild	2023-04-01 16:07:10.370453989 +0000
@@ -77,6 +77,12 @@ elif [ "$ARCH" = "x86_64" ]; then
 elif [ "$ARCH" = "armv7hl" ]; then
   SLKCFLAGS="-march=armv7-a -mfpu=vfpv3-d16"
   LIBDIRSUFFIX=""
+elif [ "$ARCH" = "aarch64" ]; then
+  SLKCFLAGS=""
+  LIBDIRSUFFIX="64"
+elif [ "$ARCH" = "riscv64" ]; then
+  SLKCFLAGS="-march=rv64gc -misa-spec=20191213"
+  LIBDIRSUFFIX="64"
 else
   SLKCFLAGS=""
   LIBDIRSUFFIX=""
@@ -128,6 +134,7 @@ autoreconf -vif
 build_grub() {
   EFI_DO="$*"
   # Configure:
+  ac_cv_header_sys_sysmacros_h=yes \
   CFLAGS="$SLKCFLAGS" \
   ./configure \
     --prefix=/usr \
