--- a/package-blacklist	2017-10-03 20:31:37.507220216 +0300
+++ b/package-blacklist	2017-10-03 20:31:19.967220216 +0300
@@ -6,6 +6,7 @@
 # They also say in the README for this one that it is not yet ready:
 xf86-video-omap
 
+
 # MacOS related packages:
 applewmproto
 windowswmproto
@@ -42,3 +43,11 @@
 # We don't want this one, as it causes failure of X with no xorg.conf
 xf86-video-fbdev
 
+# x86 related packages:
+intel-gpu-tools
+xf86-input-vmmouse
+xf86-video-geode
+xf86-video-intel
+xf86-video-siliconmotion
+xf86-video-tseng
+xf86-video-vmware
