--- a/llvm.SlackBuild	2017-12-23 12:33:22.819227755 +0000
+++ b/llvm.SlackBuild	2017-12-23 12:40:38.399227963 +0000
@@ -50,6 +50,9 @@
 elif [ "$ARCH" = "x86_64" ]; then
   SLKCFLAGS="-O2 -fPIC"
   LIBDIRSUFFIX="64"
+elif [ "$ARCH" = "aarch64" ]; then
+  SLKCFLAGS="-O2"
+  LIBDIRSUFFIX="64"
 else
   SLKCFLAGS="-O2"
   LIBDIRSUFFIX=""
@@ -109,13 +112,26 @@
 # is fixed.
 zcat $CWD/clang_disable_pgo.patch.gz | patch -d tools/clang/ -p1 --verbose || exit 1
 
+# Disable test that fails when compiled as PIE
+# https://bugs.llvm.org/show_bug.cgi?id=31870
+xzcat $CWD/disable-llvm-symbolizer-test.patch.xz | patch -Np1 --verbose || exit 1
+
+# Enable SSP and PIE by default
+xzcat $CWD/0001-GCC-compatibility-Ignore-the-fno-plt-flag.patch.xz | patch -Np1 -d tools/clang/ --verbose || exit 1
+xzcat $CWD/0002-Enable-SSP-and-PIE-by-default.patch.xz | patch -Np1 -d tools/clang/ --verbose || exit 1
+
+CCOMPILER=gcc
+CPPCOMPILER=g++
+which clang > /dev/null 2>&1 && { CCOMPILER=clang ; CPPCOMPILER=clang++ ;}
+echo "Using compiler: $CCOMPILER / $CPPCOMPILER"
+
 # need to disable assertions to make llvm thread-safe
 # clang resource dir is a relative path based on the location of the clang binary
 mkdir build
 cd build
   cmake \
-    -DCMAKE_C_COMPILER="clang" \
-    -DCMAKE_CXX_COMPILER="clang++" \
+    -DCMAKE_C_COMPILER="$CCOMPILER" \
+    -DCMAKE_CXX_COMPILER="$CPPCOMPILER" \
     -DCMAKE_C_FLAGS:STRING="$SLKCFLAGS" \
     -DCMAKE_CXX_FLAGS:STRING="$SLKCFLAGS" \
     -DCMAKE_INSTALL_PREFIX=/usr \
