--- a/mozilla-firefox.SlackBuild	2019-02-01 06:36:21.408024839 +0200
+++ b/mozilla-firefox.SlackBuild	2019-02-01 06:38:16.571015072 +0200
@@ -132,6 +132,9 @@
 elif [ "$ARCH" = "armel" ]; then
   SLKCFLAGS="-O2 -march=armv4t"
   LIBDIRSUFFIX=""
+elif [ "$ARCH" = "aarch64" ]; then
+  SLKCFLAGS="-O2"
+  LIBDIRSUFFIX="64"
 else
   SLKCFLAGS="-O2"
   LIBDIRSUFFIX=""
@@ -203,6 +206,8 @@
   \( -perm 666 -o -perm 664 -o -perm 600 -o -perm 444 -o -perm 440 -o -perm 400 \) \
   -exec chmod 644 {} \;
 
+xzcat $CWD/Bug-1238661---fix-mozillaSignalTrampoline-to-work-.patch.xz | patch -p1 --verbose || exit 1
+
 # Our building options, in a configure-like display ;)
 OPTIONS="\
   --enable-official-branding \
@@ -257,6 +262,12 @@
 # needed for PGO.
 echo ". \$topsrcdir/browser/config/mozconfig" > .mozconfig
 
+# ARM: see https://bugzilla.redhat.com/show_bug.cgi?id=1426850
+# This isn't added using $SLKCONFARGS because it doesn't cater for things with spaces in:
+# unless I edit the for loop above to use IFS with some other variable.
+# this is easier right now ;-)
+echo 'ac_add_options --enable-optimize="$SLKCFLAGS -fno-schedule-insns"' >> .mozconfig
+
 # Mozilla devs enforce using an objdir for building
 # https://developer.mozilla.org/en/Configuring_Build_Options#Building_with_an_objdir
 mkdir obj
