--- a/mozilla-firefox.SlackBuild	2018-03-14 20:36:46.315867471 +0200
+++ b/mozilla-firefox.SlackBuild	2018-03-14 20:40:57.314867471 +0200
@@ -100,6 +100,10 @@
   SLKCFLAGS="-O2 -march=armv4t"
   LIBDIRSUFFIX=""
   OPTIMIZE=${OPTIMIZE:-"-O1"}
+elif [ "$ARCH" = "aarch64" ]; then
+  SLKCFLAGS="-O1"
+  LIBDIRSUFFIX="64"
+  OPTIMIZE=" --enable-optimize='-O1 -fno-schedule-insns'" ## used to use -O1 -- this is unused. see .mozconfig below
 else
   SLKCFLAGS="-O2"
   LIBDIRSUFFIX=""
@@ -183,8 +187,33 @@
   \( -perm 666 -o -perm 664 -o -perm 600 -o -perm 444 -o -perm 440 -o -perm 400 \) \
   -exec chmod 644 {} \;
 
+# https://bugzilla.mozilla.org/show_bug.cgi?id=1314968
+xzcat $CWD/wifi-disentangle.patch.xz | patch -p1 --verbose || exit 1
+xzcat $CWD/wifi-fix-interface.patch.xz | patch -p1 --verbose || exit 1
+
+# https://bugzilla.mozilla.org/show_bug.cgi?id=1384062
+#xzcat $CWD/0001-Bug-1384062-Make-SystemResourceMonitor.stop-more-res.patch.xz | patch -p1 --verbose || exit 1
+
+# https://bugzilla.mozilla.org/show_bug.cgi?id=1382942
+xzcat $CWD/no-plt.diff.xz | patch -p1 --verbose || exit 1
+
 # Our building options, in a configure-like display ;)
 OPTIONS="\
+  --enable-system-ffi \
+  --enable-system-sqlite \
+  --enable-pie \
+  --disable-eme \
+  --disable-crashreporter \
+  --disable-updater \
+  --disable-gconf \
+  --with-system-libvpx \
+  --without-system-icu \
+  --disable-system-cairo \
+  --disable-stylo \
+  --disable-webrtc \
+  --host=$ARCH-slackware-linux \
+  --target=$ARCH-slackware-linux \
+  \
   --enable-official-branding \
   --prefix=/usr \
   --libdir=/usr/lib${LIBDIRSUFFIX} \
@@ -219,12 +248,21 @@
     browser/installer/package-manifest.in || exit 1
 fi
 
+SLKCFLAGS="$SLKCFLAGS -pipe -fstack-protector -Wformat -fexceptions -fPIC -Wl,-z,relro -Wl,-z,now"
+SLKCXXFLAGS="$SLKCFLAGS"
+SLKCPPFLAGS=""
+SLKLDFLAGS=" -Wl,--as-needed -Wl,--no-keep-memory -Wl,--stats"
+
 export MOZILLA_OFFICIAL="1"
 export BUILD_OFFICIAL="1"
 export MOZ_PHOENIX="1"
 export MOZ_PACKAGE_JSSHELL="1"
+export LDFLAGS="$SLKLDFLAGS"
+export MOZ_LINK_FLAGS="$SLKLDFLAGS"
+export MOZ_OPT_FLAGS="$SLKCFLAGS"
 export CFLAGS="$SLKCFLAGS"
-export CXXFLAGS="$SLKCFLAGS"
+export CXXFLAGS="$SLKCXXFLAGS"
+export CPPFLAGS="$SLKCPPFLAGS"
 export MOZ_MAKE_FLAGS="$NUMJOBS"
 
 # Clear some variables that could break the build
@@ -251,6 +289,12 @@
 # Write in it the options above
 for option in $OPTIONS; do echo "ac_add_options $option" >> .mozconfig; done
 
+# ARM: see https://bugzilla.redhat.com/show_bug.cgi?id=1426850
+# This isn't added using $SLKCONFARGS because it doesn't cater for things with spaces in:
+# unless I edit the for loop above to use IFS with some other variable.
+# this is easier right now ;-)
+echo 'ac_add_options --enable-optimize="-O1 -fno-schedule-insns"' >> .mozconfig
+
 # https://developer.mozilla.org/en-US/docs/Building_with_Profile-Guided_Optimization
 # Thanks to ArchLinux and Gentoo for the additional hints.
 if [ "$PGO" = "yes" ]; then
