--- a/coreutils.SlackBuild	2021-02-10 17:11:24.378222671 +0200
+++ b/coreutils.SlackBuild	2021-02-10 17:12:11.786189289 +0200
@@ -64,6 +64,12 @@
 elif [ "$ARCH" = "armel" ]; then
   SLKCFLAGS="-O2 -march=armv4t"
   LIBDIRSUFFIX=""
+elif [ "$ARCH" = "aarch64" ]; then
+  SLKCFLAGS="-O2 -fPIC"
+  LIBDIRSUFFIX="64"
+elif [ "$ARCH" = "riscv64" ]; then
+  SLKCFLAGS="-O2 -fPIC"
+  LIBDIRSUFFIX="64"
 else
   SLKCFLAGS="-O2"
   LIBDIRSUFFIX=""
@@ -93,6 +99,9 @@
 # Revert change to ls quoting style introduced in coreutils-8.25:
 zcat $CWD/no_ls_quoting.patch.gz | patch -p1 --verbose --backup --suffix=.orig || exit 1
 
+# ls: fix compilation failure on aarch64
+xzcat $CWD/coreutils-8.32-ls-aarch64.patch.xz | patch -p1 --verbose --backup --suffix=.orig || exit 1
+
 # Compilation with glibc version later than 2.3.2 needs the environment
 # variable DEFAULT_POSIX2_VERSION set to 199209.
 # Without that line, the coreutils will start complaining about 'obsolete'
