--- a/oprofile.SlackBuild	2021-08-28 19:45:51.897177190 +0000
+++ b/oprofile.SlackBuild	2021-12-13 09:00:54.209831038 +0000
@@ -58,6 +58,12 @@
 elif [ "$ARCH" = "x86_64" ]; then
   SLKCFLAGS="-O2 -fPIC"
   LIBDIRSUFFIX="64"
+elif [ "$ARCH" = "aarch64" ]; then
+  SLKCFLAGS="-O2 -fPIC"
+  LIBDIRSUFFIX="64"
+elif [ "$ARCH" = "riscv64" ]; then
+  SLKCFLAGS="-O2 -fPIC"
+  LIBDIRSUFFIX="64"
 else
   SLKCFLAGS="-O2"
   LIBDIRSUFFIX=""
@@ -82,6 +88,13 @@
   \( -perm 666 -o -perm 664 -o -perm 600 -o -perm 444 -o -perm 440 -o -perm 400 \) \
   -exec chmod 644 {} \+
 
+# supporr riscv64
+xzcat $CWD/config.guess.xz > config.guess || exit 1
+xzcat $CWD/config.sub.xz > config.sub || exit 1
+
+# adding rmb definition for riscv
+xzcat $CWD/riscv_rmb_definition.patch.xz | patch -p1 --verbose || exit 1
+
 CFLAGS="$SLKCFLAGS" \
 CXXFLAGS="$SLKCFLAGS" \
 ./configure \
