--- a/gnu-efi.SlackBuild	2021-02-13 03:26:08.431355984 +0200
+++ b/gnu-efi.SlackBuild	2021-02-13 03:26:33.811341616 +0200
@@ -57,6 +57,12 @@
 elif [ "$ARCH" = "x86_64" ]; then
   SLKCFLAGS="-O2 -fPIC"
   LIBDIRSUFFIX="64"
+elif [ "$ARCH" = "aarch64" ]; then
+  SLKCFLAGS="-O2 -fPIC"
+  LIBDIRSUFFIX="64"
+elif [ "$ARCH" = "riscv64" ]; then
+  SLKCFLAGS="-O2 -fPIC"
+  LIBDIRSUFFIX="64"
 else
   SLKCFLAGS="-O2"
   LIBDIRSUFFIX=""
@@ -74,10 +80,10 @@
 sed -i "s,/usr/local,/usr,g" Make.defaults
 
 make CPPFLAGS="$SLKCFLAGS" || exit 1
-make CPPFLAGS="$SLKCFLAGS" install INSTALLROOT=$PKG || exit
+make CPPFLAGS="$SLKCFLAGS" install INSTALLROOT=$PKG PREFIX="/usr" LIBDIR="/usr/lib${LIBDIRSUFFIX}" || exit
 
 # Fix install locations:
-mv $PKG/usr/lib $PKG/usr/lib${LIBDIRSUFFIX}
+#mv $PKG/usr/lib $PKG/usr/lib${LIBDIRSUFFIX}
 mkdir $PKG/usr/lib${LIBDIRSUFFIX}/gnuefi
 mv $PKG/usr/lib${LIBDIRSUFFIX}/*.lds $PKG/usr/lib${LIBDIRSUFFIX}/gnuefi
 mv $PKG/usr/lib${LIBDIRSUFFIX}/crt0*.o $PKG/usr/lib${LIBDIRSUFFIX}/gnuefi
