--- configure.orig      2009-09-05 16:09:50.238987776 +0100
+++ configure   2009-09-05 16:10:15.418985020 +0100
@@ -38409,6 +38409,7 @@
 case "${build_vendor}" in
 unknown|pc|ibm|redhat|pld|mandrake|conectiva|lvr|yellowdog|caos|suse)
        test -f /etc/redhat-release &&          RPMCANONVENDOR=redhat
+       test -f /etc/slackware-version &&       RPMCANONVENDOR=rpm
        test -f /etc/SuSE-release &&            RPMCANONVENDOR=suse
        test -f /etc/pld-release &&             RPMCANONVENDOR=pld
        test -f /etc/mandrake-release &&        RPMCANONVENDOR=mandrake

