--- openssh-6.7p1.orig/serverloop.c	2014-08-19 02:14:17.000000000 +0100
+++ openssh-6.7p1/serverloop.c	2015-11-02 13:09:57.659914304 +0000
@@ -957,6 +957,7 @@
 	Channel *c = NULL;
 	char *target, *originator;
 	u_short target_port, originator_port;
+	uid_t who;
 
 	target = packet_get_string(NULL);
 	target_port = packet_get_int();
@@ -964,8 +965,11 @@
 	originator_port = packet_get_int();
 	packet_check_eom();
 
-	debug("server_request_direct_tcpip: originator %s port %d, target %s "
-	    "port %d", originator, originator_port, target, target_port);
+//	debug("server_request_direct_tcpip: originator %s port %d, target %s "
+//	    "port %d", originator, originator_port, target, target_port);
+	who = getuid();
+	logit("Tunnel: %s:%d -> %s:%d UID(%d)",
+	    originator, originator_port, target, target_port, who);
 
 	/* XXX fine grained permissions */
 	if ((options.allow_tcp_forwarding & FORWARD_LOCAL) != 0 &&
