--- a/gcc.SlackBuild	2018-04-23 17:27:12.429104534 +0100
+++ /bgcc.SlackBuild	2018-04-23 17:53:09.034105277 +0100
@@ -114,6 +114,10 @@
   SLKCFLAGS="-O3 -march=armv7-a -mfpu=vfpv3-d16"
   LIBDIRSUFFIX=""
   LIB_ARCH=armv7hl
+elif [ "$ARCH" = "aarch64" ]; then
+  SLKCFLAGS="-O2"
+  LIBDIRSUFFIX="64"
+  LIB_ARCH=aarch64
 else
   SLKCFLAGS="-O2"
   LIBDIRSUFFIX=""
@@ -133,11 +137,11 @@
 # These are the directories to build other packages in:
 PKG2=$TMP/package-gcc-g++
 PKG3=$TMP/package-gcc-gfortran
-PKG4=$TMP/package-gcc-gnat
+#PKG4=$TMP/package-gcc-gnat
 PKG6=$TMP/package-gcc-objc
 #PKG7=$TMP/package-gcc-g++-gch
 PKG8=$TMP/package-gcc-go
-PKG9=$TMP/package-gcc-brig
+#PKG9=$TMP/package-gcc-brig
 
 # Clear the build locations:
 rm -rf $PKG{1,2,3,4,6,8,9}
@@ -148,11 +152,11 @@
 cat $CWD/slack-desc.gcc > $PKG1/install/slack-desc
 cat $CWD/slack-desc.gcc-g++ > $PKG2/install/slack-desc
 cat $CWD/slack-desc.gcc-gfortran > $PKG3/install/slack-desc
-cat $CWD/slack-desc.gcc-gnat > $PKG4/install/slack-desc
+#cat $CWD/slack-desc.gcc-gnat > $PKG4/install/slack-desc
 cat $CWD/slack-desc.gcc-objc > $PKG6/install/slack-desc
 #cat $CWD/slack-desc.gcc-g++-gch > $PKG7/install/slack-desc
 cat $CWD/slack-desc.gcc-go > $PKG8/install/slack-desc
-cat $CWD/slack-desc.gcc-brig > $PKG9/install/slack-desc
+#cat $CWD/slack-desc.gcc-brig > $PKG9/install/slack-desc
 
 cd $TMP
 rm -rf gcc-$VERSION
@@ -204,16 +208,16 @@
       fi
     )
 
-    mkdir -p $PKG4/usr/doc/gcc-${VERSION}/gcc/ada
-    ( cd ada || exit 0
-      cp -a \
-        ChangeLog.tree-ssa \
-      $PKG4/usr/doc/gcc-$VERSION/gcc/ada
-      if [ -r ChangeLog ]; then
-        cat ChangeLog | head -n 1000 > $PKG4/usr/doc/gcc-$VERSION/gcc/ada/ChangeLog
-        touch -r ChangeLog $PKG4/usr/doc/gcc-$VERSION/gcc/ada/ChangeLog
-      fi
-    )
+#    mkdir -p $PKG4/usr/doc/gcc-${VERSION}/gcc/ada
+#    ( cd ada || exit 0
+#      cp -a \
+#        ChangeLog.tree-ssa \
+#      $PKG4/usr/doc/gcc-$VERSION/gcc/ada
+#      if [ -r ChangeLog ]; then
+#        cat ChangeLog | head -n 1000 > $PKG4/usr/doc/gcc-$VERSION/gcc/ada/ChangeLog
+#        touch -r ChangeLog $PKG4/usr/doc/gcc-$VERSION/gcc/ada/ChangeLog
+#      fi
+#    )
 
     mkdir -p $PKG6/usr/doc/gcc-${VERSION}/gcc/objc
     ( cd objc || exit 0
@@ -238,13 +242,13 @@
       $PKG8/usr/doc/gcc-${VERSION}/gcc/go
     )
 
-    mkdir -p $PKG9/usr/doc/gcc-${VERSION}/gcc/brig
-    ( cd brig || exit 0
-      if [ -r ChangeLog ]; then
-        cat ChangeLog | head -n 1000 > $PKG9/usr/doc/gcc-${VERSION}/gcc/brig/ChangeLog
-        touch -r ChangeLog $PKG9/usr/doc/gcc-${VERSION}/gcc/brig/ChangeLog
-      fi
-    )
+#    mkdir -p $PKG9/usr/doc/gcc-${VERSION}/gcc/brig
+#    ( cd brig || exit 0
+#      if [ -r ChangeLog ]; then
+#        cat ChangeLog | head -n 1000 > $PKG9/usr/doc/gcc-${VERSION}/gcc/brig/ChangeLog
+#        touch -r ChangeLog $PKG9/usr/doc/gcc-${VERSION}/gcc/brig/ChangeLog
+#      fi
+#    )
 
   ) || exit 1
 
@@ -256,13 +260,13 @@
     fi
   )
 
-  mkdir -p $PKG4/usr/doc/gcc-${VERSION}/libada
-  ( cd libada || exit 0
-    if [ -r ChangeLog ]; then
-      cat ChangeLog | head -n 1000 > $PKG4/usr/doc/gcc-${VERSION}/libada/ChangeLog
-      touch -r ChangeLog $PKG4/usr/doc/gcc-${VERSION}/libada/ChangeLog
-    fi
-  )
+#  mkdir -p $PKG4/usr/doc/gcc-${VERSION}/libada
+#  ( cd libada || exit 0
+#    if [ -r ChangeLog ]; then
+#      cat ChangeLog | head -n 1000 > $PKG4/usr/doc/gcc-${VERSION}/libada/ChangeLog
+#      touch -r ChangeLog $PKG4/usr/doc/gcc-${VERSION}/libada/ChangeLog
+#    fi
+#  )
 
   mkdir -p $PKG1/usr/doc/gcc-${VERSION}/libgomp
   ( cd libgomp || exit 0
@@ -308,12 +312,14 @@
   # Correct me if my take on this is wrong.
   #  --with-cpu=$ARCH 
  
-  if [ "$ARCH" != "x86_64" ]; then
-    GCC_ARCHOPTS="--with-arch=$ARCH"
-  else
-    GCC_ARCHOPTS="--disable-multilib"
-  fi
- 
+#  if [ "$ARCH" != "x86_64" ]; then
+#    GCC_ARCHOPTS="--with-arch=$ARCH"
+#  else
+#    GCC_ARCHOPTS="--disable-multilib"
+#  fi
+
+  [[ "$ARCH" == aarch64 ]] && GCC_ARCHOPTS="--disable-multilib"
+
   CFLAGS="$SLKCFLAGS" \
   CXXFLAGS="$SLKCFLAGS" \
   ../gcc-$VERSION/configure --prefix=/usr \
@@ -322,7 +328,7 @@
      --infodir=/usr/info \
      --enable-shared \
      --enable-bootstrap \
-     --enable-languages=ada,brig,c,c++,fortran,go,lto,objc \
+     --enable-languages=c,c++,fortran,go,lto,objc \
      --enable-threads=posix \
      --enable-checking=release \
      --enable-objc-gc \
@@ -348,15 +354,15 @@
   # Include all debugging info (for now):
   make $NUMJOBS bootstrap || exit 1
 
-  ( cd gcc
-    make $NUMJOBS gnatlib GNATLIBCFLAGS="$SLKCFLAGS" || exit 1
+#  ( cd gcc
+#    make $NUMJOBS gnatlib GNATLIBCFLAGS="$SLKCFLAGS" || exit 1
     # This wants a shared -ladd2line?
     #make gnatlib-shared || exit 1
     
-    CFLAGS="$SLKCFLAGS" \
-    CXXFLAGS="$SLKCFLAGS" \
-    make $NUMJOBS gnattools || exit 1
-  ) || exit 1
+#    CFLAGS="$SLKCFLAGS" \
+#    CXXFLAGS="$SLKCFLAGS" \
+#    make $NUMJOBS gnattools || exit 1
+#  ) || exit 1
   make info || exit 1
 
   # Set GCCCHECK=something to run the tests
@@ -507,18 +513,18 @@
 )
 
 # gcc-gnat:
-( cd $PKG4
-  mkdir -p usr/bin
-  mv $PKG1/usr/bin/gnat* usr/bin
-  mv $PKG1/usr/bin/gpr* usr/bin
-  mkdir -p usr/info
-  mv $PKG1/usr/info/gnat* usr/info
-  mkdir -p usr/libexec/gcc/$TARGET/$VERSION
-  mv $PKG1/usr/libexec/gcc/$TARGET/$VERSION/gnat1 usr/libexec/gcc/$TARGET/$VERSION
-  mkdir -p usr/lib${LIBDIRSUFFIX}/gcc/$TARGET/$VERSION
-  mv $PKG1/usr/lib${LIBDIRSUFFIX}/gcc/$TARGET/$VERSION/adainclude usr/lib${LIBDIRSUFFIX}/gcc/$TARGET/$VERSION
-  mv $PKG1/usr/lib${LIBDIRSUFFIX}/gcc/$TARGET/$VERSION/adalib usr/lib${LIBDIRSUFFIX}/gcc/$TARGET/$VERSION
-)
+#( cd $PKG4
+#  mkdir -p usr/bin
+#  mv $PKG1/usr/bin/gnat* usr/bin
+#  mv $PKG1/usr/bin/gpr* usr/bin
+#  mkdir -p usr/info
+#  mv $PKG1/usr/info/gnat* usr/info
+#  mkdir -p usr/libexec/gcc/$TARGET/$VERSION
+#  mv $PKG1/usr/libexec/gcc/$TARGET/$VERSION/gnat1 usr/libexec/gcc/$TARGET/$VERSION
+#  mkdir -p usr/lib${LIBDIRSUFFIX}/gcc/$TARGET/$VERSION
+#  mv $PKG1/usr/lib${LIBDIRSUFFIX}/gcc/$TARGET/$VERSION/adainclude usr/lib${LIBDIRSUFFIX}/gcc/$TARGET/$VERSION
+#  mv $PKG1/usr/lib${LIBDIRSUFFIX}/gcc/$TARGET/$VERSION/adalib usr/lib${LIBDIRSUFFIX}/gcc/$TARGET/$VERSION
+#)
 
 # gcc-objc:
 ( cd $PKG6
@@ -562,14 +568,14 @@
 ) || exit 1
 
 # gcc-brig:
-( cd $PKG9
-  mkdir -p usr/bin
-  mv $PKG1/usr/bin/*brig* usr/bin
-  mkdir -p usr/lib${LIBDIRSUFFIX}
-  mv $PKG1/usr/lib${LIBDIRSUFFIX}/libhsail* usr/lib${LIBDIRSUFFIX}
-  mkdir -p usr/libexec/gcc/$TARGET/$VERSION
-  mv $PKG1/usr/libexec/gcc/$TARGET/$VERSION/brig1 usr/libexec/gcc/$TARGET/$VERSION
-)
+#( cd $PKG9
+#  mkdir -p usr/bin
+#  mv $PKG1/usr/bin/*brig* usr/bin
+#  mkdir -p usr/lib${LIBDIRSUFFIX}
+#  mv $PKG1/usr/lib${LIBDIRSUFFIX}/libhsail* usr/lib${LIBDIRSUFFIX}
+#  mkdir -p usr/libexec/gcc/$TARGET/$VERSION
+#  mv $PKG1/usr/libexec/gcc/$TARGET/$VERSION/brig1 usr/libexec/gcc/$TARGET/$VERSION
+#)
 
 # Generate packages:
 ( cd $PKG1
@@ -578,14 +584,14 @@
   makepkg -l y -c n $TMP/gcc-g++-$VERSION-$ARCH-$BUILD.txz )
 ( cd $PKG3
   makepkg -l y -c n $TMP/gcc-gfortran-$VERSION-$ARCH-$BUILD.txz )
-( cd $PKG4
-  makepkg -l y -c n $TMP/gcc-gnat-$VERSION-$ARCH-$BUILD.txz )
+#( cd $PKG4
+#  makepkg -l y -c n $TMP/gcc-gnat-$VERSION-$ARCH-$BUILD.txz )
 ( cd $PKG6
   makepkg -l y -c n $TMP/gcc-objc-$VERSION-$ARCH-$BUILD.txz )
 ( cd $PKG8
   makepkg -l y -c n $TMP/gcc-go-$VERSION-$ARCH-$BUILD.txz )
-( cd $PKG9
-  makepkg -l y -c n $TMP/gcc-brig-$VERSION-$ARCH-$BUILD.txz )
+#( cd $PKG9
+#  makepkg -l y -c n $TMP/gcc-brig-$VERSION-$ARCH-$BUILD.txz )
 
 echo
 echo "Slackware GCC package build complete!"
