--- a/acct.SlackBuild	2017-09-17 14:42:06.604233635 +0100
+++ b/acct.SlackBuild	2017-09-17 14:43:34.304233677 +0100
@@ -35,16 +35,20 @@
   esac
 fi
 
-if [ "$ARCH" = "i586" ]; then
-  SLKCFLAGS="-O2 -march=i586 -mtune=i686"
-  LIBDIRSUFFIX=""
-elif [ "$ARCH" = "s390" ]; then
-  SLKCFLAGS="-O2"
-  LIBDIRSUFFIX=""
-elif [ "$ARCH" = "x86_64" ]; then
-  SLKCFLAGS="-O2 -fPIC"
-  LIBDIRSUFFIX="64"
-fi
+case "$ARCH" in
+  i?86)    SLKCFLAGS="-O2 -march=i586 -mtune=i686"
+           LIBDIRSUFFIX=""
+           ;;
+  s390)    SLKCFLAGS="-O2"
+           LIBDIRSUFFIX=""
+           ;;
+  x86_64)  SLKCFLAGS="-O2 -fPIC"
+           LIBDIRSUFFIX="64"
+           ;;
+  aarch64) SLKCFLAGS="-O2"
+           LIBDIRSUFFIX="64"
+           ;;
+esac
 
 CWD=$(pwd)
 TMP=${TMP:-/tmp}
