--- a/tix.SlackBuild	2017-09-25 20:22:01.362572950 +0100
+++ b/tix.SlackBuild	2017-09-25 20:23:12.578572984 +0100
@@ -46,6 +46,9 @@
 elif [ "$ARCH" = "x86_64" ]; then
   SLKCFLAGS="-O2 -fPIC"
   LIBDIRSUFFIX="64"
+elif [ "$ARCH" = "aarch64" ]; then
+  SLKCFLAGS="-O2"
+  LIBDIRSUFFIX="64"
 fi
 
 CWD=$(pwd)
@@ -65,8 +68,13 @@
   \( -perm 666 -o -perm 664 -o -perm 600 -o -perm 444 -o -perm 440 -o -perm 400 \) \
   -exec chmod 644 {} \;
 
+autoconf || exit 1
+
 zcat $CWD/tix.tcl8.5.patch.gz | patch -p1 --backup --suffix=.orig || exit 1
 zcat $CWD/tix.soname.patch.gz | patch -p1 || exit 1
+xzcat $CWD/tix-8.4.3-tcl86.patch.xz | patch -p1 || exit 1
+sed -i -e 's:generic/tclInt.h:tclInt.h:g' configure
+sed -i -e 's:generic/tkInt.h:tkInt.h:g' configure
 
 CFLAGS="$SLKCFLAGS" \
 CPPFLAGS="-DUSE_INTERP_RESULT" \
