--- netkit-rwho-0.17/rwho/Makefile.strip	Wed Jun 19 17:33:10 2002
+++ netkit-rwho-0.17/rwho/Makefile	Wed Jun 19 17:33:41 2002
@@ -9,7 +9,7 @@
 rwho.o: ../version.h
 
 install: rwho
-	install -s -m$(BINMODE) rwho $(INSTALLROOT)$(BINDIR)
+	install -m$(BINMODE) rwho $(INSTALLROOT)$(BINDIR)
 	install -m$(MANMODE) rwho.1 $(INSTALLROOT)$(MANDIR)/man1/
 
 clean:
--- netkit-rwho-0.17/ruptime/Makefile.strip	Wed Jun 19 17:33:10 2002
+++ netkit-rwho-0.17/ruptime/Makefile	Wed Jun 19 17:33:56 2002
@@ -7,7 +7,7 @@
 	$(CC) $(LDFLAGS) $^ $(LIBS) -o $@
 
 install: ruptime
-	install -s -m$(BINMODE) ruptime $(INSTALLROOT)$(BINDIR)
+	install -m$(BINMODE) ruptime $(INSTALLROOT)$(BINDIR)
 	install -m$(MANMODE) ruptime.1 $(INSTALLROOT)$(MANDIR)/man1/
 
 clean:
--- netkit-rwho-0.17/rwhod/Makefile.strip	Wed Jun 19 17:33:10 2002
+++ netkit-rwho-0.17/rwhod/Makefile	Wed Jun 19 17:34:10 2002
@@ -11,7 +11,7 @@
 $(OBJS): daemon.h
 
 install: rwhod
-	install -s -m$(DAEMONMODE) rwhod $(INSTALLROOT)$(SBINDIR)
+	install -m$(DAEMONMODE) rwhod $(INSTALLROOT)$(SBINDIR)
 	install -m$(MANMODE) rwhod.8 $(INSTALLROOT)$(MANDIR)/man8/
 
 clean:
