--- a/mozilla-nss.SlackBuild	2020-06-17 19:27:48.960955198 +0000
+++ b/mozilla-nss.SlackBuild	2020-06-18 17:03:11.808725653 +0000
@@ -60,6 +60,10 @@
 elif [ "$ARCH" = "armv7hl" ]; then
   SLKCFLAGS="-O2 -march=armv7-a -mfpu=vfpv3-d16"
   LIBDIRSUFFIX=""
+elif [ "$ARCH" = "aarch64" ]; then
+  SLKCFLAGS="-O2"
+  LIBDIRSUFFIX="64"
+  export USE_64=1
 else
   SLKCFLAGS="-O2"
   LIBDIRSUFFIX=""
@@ -68,7 +72,7 @@
 TMP=${TMP:-/tmp}
 PKG=$TMP/package-$PKGNAM
 
-NUMJOBS=${NUMJOBS:-" -j$(expr $(nproc) + 1) "}
+NUMJOBS=${NUMJOBS:-" -j $(expr $(nproc) + 1) "}
 
 rm -rf $PKG
 mkdir -p $TMP $PKG
